{
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "setwd(\"/Users/sfchen/Documents/2018_Su/Applied-Bioinformatics-HW/Unit2-module2-ChIPseq/aligned_bam\")\n",
        "\n",
        "# Load ChIPQC package\n",
        "library(ChIPQC)"
      ],
      "outputs": [],
      "execution_count": 2,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "code",
      "source": [
        "samples <- read.csv(\"ChIPQC_samplesheet_chr2.chr11.csv\")\n",
        "head(samples)"
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Warning message in read.table(file = file, header = header, sep = sep, quote = quote, :\n",
            "“incomplete final line found by readTableHeader on 'ChIPQC_samplesheet_chr2.chr11.csv'”"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<table>\n",
              "<thead><tr><th scope=col>SampleID</th><th scope=col>Tissue</th><th scope=col>Factor</th><th scope=col>Condition</th><th scope=col>Replicate</th><th scope=col>bamReads</th><th scope=col>ControlID</th><th scope=col>bamControl</th><th scope=col>Peaks</th><th scope=col>PeakCaller</th></tr></thead>\n",
              "<tbody>\n",
              "\t<tr><td>SRR3001750_chr2.chr11                             </td><td>NA                                                </td><td>NA                                                </td><td>NA                                                </td><td>NA                                                </td><td>SRR3001750_srt_dupr.chr2.chr11.srt.bam            </td><td>NA                                                </td><td>NA                                                </td><td>SRR3001750_srt_dupr.chr2.chr11.srt_peaks.broadPeak</td><td>macs                                              </td></tr>\n",
              "</tbody>\n",
              "</table>\n"
            ],
            "text/markdown": [
              "\n",
              "SampleID | Tissue | Factor | Condition | Replicate | bamReads | ControlID | bamControl | Peaks | PeakCaller | \n",
              "|---|\n",
              "| SRR3001750_chr2.chr11                              | NA                                                 | NA                                                 | NA                                                 | NA                                                 | SRR3001750_srt_dupr.chr2.chr11.srt.bam             | NA                                                 | NA                                                 | SRR3001750_srt_dupr.chr2.chr11.srt_peaks.broadPeak | macs                                               | \n",
              "\n\n"
            ],
            "text/latex": [
              "\\begin{tabular}{r|llllllllll}\n",
              " SampleID & Tissue & Factor & Condition & Replicate & bamReads & ControlID & bamControl & Peaks & PeakCaller\\\\\n",
              "\\hline\n",
              "\t SRR3001750\\_chr2.chr11                                  & NA                                                       & NA                                                       & NA                                                       & NA                                                       & SRR3001750\\_srt\\_dupr.chr2.chr11.srt.bam               & NA                                                       & NA                                                       & SRR3001750\\_srt\\_dupr.chr2.chr11.srt\\_peaks.broadPeak & macs                                                    \\\\\n",
              "\\end{tabular}\n"
            ],
            "text/plain": [
              "  SampleID              Tissue Factor Condition Replicate\n",
              "1 SRR3001750_chr2.chr11 NA     NA     NA        NA       \n",
              "  bamReads                               ControlID bamControl\n",
              "1 SRR3001750_srt_dupr.chr2.chr11.srt.bam NA        NA        \n",
              "  Peaks                                              PeakCaller\n",
              "1 SRR3001750_srt_dupr.chr2.chr11.srt_peaks.broadPeak macs      "
            ]
          },
          "metadata": {}
        }
      ],
      "execution_count": 4,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "code",
      "source": [
        "chipObj <- ChIPQC(samples, annotation=\"mm10\") "
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "SRR3001750_chr2.chr11     NA macs\n",
            "Checking chromosomes:\n",
            "Compiling annotation...\n",
            "Computing metrics for 1 samples...\n",
            "list\n",
            "Warning message in if (is.na(peaks)) peaks = NULL:\n",
            "“條件的長度 > 1，因此只能用其第一元素”Bam file has 22 contigs\n",
            "Calculating coverage histogram for chr11\n",
            "\n",
            "Calculating SSD for chr11\n",
            "\n",
            "Calculating unique positions per strand for chr11\n",
            "\n",
            "Calculating shift for chr11\n",
            "\n",
            "Counting reads in features for chr11\n",
            "\n",
            "Signal over peaks for chr11\n",
            "\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] \"chr11\"\n",
            " 300 / 300\n",
            "done\n",
            "Calculating coverage\n",
            "Calculating Summits on  chr11  ..[1] 1\n"
          ]
        }
      ],
      "execution_count": 10,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "code",
      "source": [
        "str(chipObj)"
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Formal class 'ChIPQCexperiment' [package \"ChIPQC\"] with 4 slots\n",
            "  ..@ .Data     : list()\n",
            "  ..@ Samples   :List of 1\n",
            "  .. ..$ SRR3001750_chr2.chr11:Formal class 'ChIPQCsample' [package \"ChIPQC\"] with 19 slots\n",
            "  .. .. .. ..@ AveragePeakSignal:List of 2\n",
            "  .. .. .. .. ..$ : num [1:400] 0.469 0.471 0.474 0.475 0.478 ...\n",
            "  .. .. .. .. ..$ : num [1:400] 0.64 0.643 0.646 0.648 0.652 ...\n",
            "  .. .. .. ..@ CrossCoverage    : num [1:300] 774699 774770 774758 774731 774674 ...\n",
            "  .. .. .. ..@ CrossCorrelation : num NA\n",
            "  .. .. .. ..@ SSD              : num 0.847\n",
            "  .. .. .. ..@ SSDBL            : num NA\n",
            "  .. .. .. ..@ CountsInPeaks    : int 482251\n",
            "  .. .. .. ..@ CountsInBlackList: num NA\n",
            "  .. .. .. ..@ CountsInFeatures :List of 9\n",
            "  .. .. .. .. ..$ Peaks                  : int 482251\n",
            "  .. .. .. .. ..$ LongPromoter20000to2000: int 234182\n",
            "  .. .. .. .. ..$ Promoters2000to500     : int 34398\n",
            "  .. .. .. .. ..$ Promoters500           : int 14637\n",
            "  .. .. .. .. ..$ All5utrs               : int 10430\n",
            "  .. .. .. .. ..$ Alltranscripts         : int 480022\n",
            "  .. .. .. .. ..$ Allcds                 : int 38825\n",
            "  .. .. .. .. ..$ Allintrons             : int 439487\n",
            "  .. .. .. .. ..$ All3utrs               : int 21125\n",
            "  .. .. .. ..@ PropInFeatures   :List of 8\n",
            "  .. .. .. .. ..$ LongPromoter20000to2000: num 0.255\n",
            "  .. .. .. .. ..$ Promoters2000to500     : num 0.0232\n",
            "  .. .. .. .. ..$ Promoters500           : num 0.00786\n",
            "  .. .. .. .. ..$ All5utrs               : num 0.00461\n",
            "  .. .. .. .. ..$ Alltranscripts         : num 0.479\n",
            "  .. .. .. .. ..$ Allcds                 : num 0.0217\n",
            "  .. .. .. .. ..$ Allintrons             : num 0.44\n",
            "  .. .. .. .. ..$ All3utrs               : num 0.0184\n",
            "  .. .. .. ..@ CoverageHistogram: Named num [1:52] 2.66e+09 2.07e+07 4.73e+06 1.30e+06 4.28e+05 ...\n",
            "  .. .. .. .. ..- attr(*, \"names\")= chr [1:52] \"0\" \"1\" \"2\" \"3\" ...\n",
            "  .. .. .. ..@ FlagAndTagCounts : Named num [1:6] 0 778162 0 733720 0 ...\n",
            "  .. .. .. .. ..- attr(*, \"names\")= chr [1:6] \"UnMapped\" \"Mapped\" \"Duplicates\" \"MapQPass\" ...\n",
            "  .. .. .. ..@ readlength       : num 48\n",
            "  .. .. .. ..@ seqnames         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 84922\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges           :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:84922] 3101133 3102774 3103705 3104571 3106853 3112273 3113637 3114494 3115464 3117596 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:84922] 1293 527 415 226 150 800 215 158 188 148 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand           :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 3\n",
            "  .. .. .. .. .. ..@ lengths        : int 84922\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo          :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int NA\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr NA\n",
            "  .. .. .. ..@ elementMetadata  :Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 84922\n",
            "  .. .. .. .. .. ..@ listData       :List of 2\n",
            "  .. .. .. .. .. .. ..$ Counts              : int [1:84922] 180 50 60 31 14 12 4 6 9 4 ...\n",
            "  .. .. .. .. .. .. ..$ bedRangesSummitsTemp: num [1:84922] 3101765 3102983 3103983 3104718 3106968 ...\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType      : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata         : list()\n",
            "  ..@ DBA       :List of 12\n",
            "  .. ..$ config     :List of 11\n",
            "  .. .. ..$ RunParallel    : logi TRUE\n",
            "  .. .. ..$ reportInit     : chr \"DBA\"\n",
            "  .. .. ..$ DataType       : num 2\n",
            "  .. .. ..$ AnalysisMethod : Factor w/ 1 level \"DESeq2\": 1\n",
            "  .. .. ..$ minQCth        : num 15\n",
            "  .. .. ..$ fragmentSize   : Named num 175\n",
            "  .. .. .. ..- attr(*, \"names\")= chr \"SRR3001750_chr2.chr11\"\n",
            "  .. .. ..$ bCorPlot       : logi FALSE\n",
            "  .. .. ..$ th             : num 0.05\n",
            "  .. .. ..$ bUsePval       : logi FALSE\n",
            "  .. .. ..$ parallelPackage: num 1\n",
            "  .. .. ..$ mapQCth        : num 15\n",
            "  .. ..$ chrmap     : chr [1:2] \"chr11\" \"chr2\"\n",
            "  .. ..$ peaks      :List of 1\n",
            "  .. .. ..$ :'data.frame':\t217554 obs. of  4 variables:\n",
            "  .. .. .. ..$ chr11   : chr [1:217554] \"chr11\" \"chr11\" \"chr11\" \"chr11\" ...\n",
            "  .. .. .. ..$ X3100344: int [1:217554] 3101133 3102774 3103705 3104571 3106853 3112273 3113637 3114494 3115464 3117596 ...\n",
            "  .. .. .. ..$ X3100931: int [1:217554] 3102425 3103300 3104119 3104796 3107002 3113072 3113851 3114651 3115651 3117743 ...\n",
            "  .. .. .. ..$ X3.95704: num [1:217554] 0.206 0.143 0.196 0.137 0.144 ...\n",
            "  .. ..$ class      : chr [1:12, 1] \"SRR3001750_chr2.chr11\" \"\" \"\" \"\" ...\n",
            "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
            "  .. .. .. ..$ : chr [1:12] \"ID\" \"Tissue\" \"Factor\" \"Condition\" ...\n",
            "  .. .. .. ..$ : chr \"SRR3001750_chr2.chr11\"\n",
            "  .. ..$ masks      :List of 7\n",
            "  .. .. ..$     : logi TRUE\n",
            "  .. .. ..$     : logi TRUE\n",
            "  .. .. ..$     : logi TRUE\n",
            "  .. .. ..$     : logi TRUE\n",
            "  .. .. ..$ macs: logi TRUE\n",
            "  .. .. ..$ All : logi TRUE\n",
            "  .. .. ..$ None: logi FALSE\n",
            "  .. ..$ samples    :'data.frame':\t1 obs. of  11 variables:\n",
            "  .. .. ..$ SampleID  : Factor w/ 1 level \"SRR3001750_chr2.chr11\": 1\n",
            "  .. .. ..$ Tissue    : chr \"\"\n",
            "  .. .. ..$ Factor    : chr \"\"\n",
            "  .. .. ..$ Condition : chr \"\"\n",
            "  .. .. ..$ Replicate : chr \"\"\n",
            "  .. .. ..$ bamReads  : Factor w/ 1 level \"SRR3001750_srt_dupr.chr2.chr11.srt.bam\": 1\n",
            "  .. .. ..$ ControlID : logi NA\n",
            "  .. .. ..$ bamControl: logi NA\n",
            "  .. .. ..$ Peaks     : Factor w/ 1 level \"SRR3001750_srt_dupr.chr2.chr11.srt_peaks.broadPeak\": 1\n",
            "  .. .. ..$ PeakCaller: Factor w/ 1 level \"macs\": 1\n",
            "  .. .. ..$ Treatment : chr \"\"\n",
            "  .. ..$ called     : num [1:217554, 1] 1 1 1 1 1 1 1 1 1 1 ...\n",
            "  .. ..$ totalMerged: int 217554\n",
            "  .. ..$ merged     : num [1:217554, 1:3] 1 1 1 1 1 1 1 1 1 1 ...\n",
            "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
            "  .. .. .. ..$ : chr [1:217554] \"1\" \"2\" \"3\" \"4\" ...\n",
            "  .. .. .. ..$ : chr [1:3] \"CHR\" \"START\" \"END\"\n",
            "  .. ..$ binding    : num [1:217554, 1:4] 1 1 1 1 1 1 1 1 1 1 ...\n",
            "  .. .. ..- attr(*, \"dimnames\")=List of 2\n",
            "  .. .. .. ..$ : chr [1:217554] \"1\" \"2\" \"3\" \"4\" ...\n",
            "  .. .. .. ..$ : chr [1:4] \"CHR\" \"START\" \"END\" \"SRR3001750_chr2.chr11\"\n",
            "  .. ..$ attributes : num 1\n",
            "  .. ..$ minOverlap : num 2\n",
            "  .. ..- attr(*, \"class\")= chr \"DBA\"\n",
            "  ..@ annotation:List of 9\n",
            "  .. ..$ version                : chr \"mm10\"\n",
            "  .. ..$ LongPromoter20000to2000:Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 1505\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:1505] 3104021 3182353 3269131 3310731 3432437 3468227 3494702 3629494 3875240 3943975 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:1505] 18000 18000 18000 18000 18000 18000 18000 18000 18000 18000 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 746 759\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 1505\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ Promoters2000to500     :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 1882\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:1882] 3122021 3200353 3287131 3328731 3450437 3486227 3512702 3647494 3893240 3961975 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:1882] 1500 1500 1500 1500 1500 1500 1500 1500 1500 1500 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 933 949\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 1882\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ Promoters500           :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 1900\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:1900] 3123521 3201853 3288631 3330231 3451937 3487727 3514202 3648994 3894740 3963475 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:1900] 500 500 500 500 500 500 500 500 500 500 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 945 955\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 1900\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ All5utrs               :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 3412\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:3412] 3202353 3202583 3202995 3203630 3224632 3289131 3289869 3293280 3294456 3299034 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:3412] 204 215 332 28 176 217 745 64 62 108 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 1665 1747\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 3412\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ Alltranscripts         :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 1953\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:1953] 3124021 3202353 3289131 3330731 3452437 3488227 3514702 3649494 3895240 3963975 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:1953] 7924 42236 31592 12241 27395 5940 2650 40879 6057 16030 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 992 961\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 1953\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ Allcds                 :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 15378\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:15378] 3203658 3213260 3215592 3220244 3221093 3224632 3227395 3227706 3229869 3233988 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:15378] 93 74 128 404 202 176 136 180 311 72 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 7459 7919\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 15378\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ Allintrons             :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 14256\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:14256] 3124125 3126059 3127645 3129166 3129521 3129912 3202557 3203751 3213334 3215720 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:14256] 1663 1420 1284 216 244 241 1073 9509 2258 4524 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 6971 7285\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 14256\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n",
            "  .. ..$ All3utrs               :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
            "  .. .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 1 level \"chr11\": 1\n",
            "  .. .. .. .. .. ..@ lengths        : int 2064\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ start          : int [1:2064] 3220648 3230180 3244090 3291889 3299158 3307648 3333636 3341728 3493184 3516383 ...\n",
            "  .. .. .. .. .. ..@ width          : int [1:2064] 31 7 499 3541 48 1436 327 1244 983 969 ...\n",
            "  .. .. .. .. .. ..@ NAMES          : NULL\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2\n",
            "  .. .. .. .. .. ..@ lengths        : int [1:2] 1001 1063\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
            "  .. .. .. .. .. ..@ seqnames   : chr \"chr11\"\n",
            "  .. .. .. .. .. ..@ seqlengths : int 122082543\n",
            "  .. .. .. .. .. ..@ is_circular: logi NA\n",
            "  .. .. .. .. .. ..@ genome     : chr \"mm10\"\n",
            "  .. .. .. ..@ elementMetadata:Formal class 'DataFrame' [package \"S4Vectors\"] with 6 slots\n",
            "  .. .. .. .. .. ..@ rownames       : NULL\n",
            "  .. .. .. .. .. ..@ nrows          : int 2064\n",
            "  .. .. .. .. .. ..@ listData       : Named list()\n",
            "  .. .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. .. .. ..@ elementMetadata: NULL\n",
            "  .. .. .. .. .. ..@ metadata       : list()\n",
            "  .. .. .. ..@ elementType    : chr \"ANY\"\n",
            "  .. .. .. ..@ metadata       : list()\n"
          ]
        }
      ],
      "execution_count": 11,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "code",
      "source": [
        "ChIPQCreport(chipObj, reportName=\"ChIP.QC.report.chr2.ch11\", reportFolder=\"ChIPQCreportChr2.chr11\")"
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Saving 7 x 7 in image\n",
            "Saving 7 x 7 in image\n"
          ]
        },
        {
          "output_type": "error",
          "ename": "ERROR",
          "evalue": "Error in data.frame(Sample = meltedDF$Var2, GenomicIntervals = meltedDF$Var1, : arguments imply differing number of rows: 0, 8\n",
          "traceback": [
            "Error in data.frame(Sample = meltedDF$Var2, GenomicIntervals = meltedDF$Var1, : arguments imply differing number of rows: 0, 8\nTraceback:\n",
            "1. ChIPQCreport(chipObj, reportName = \"ChIP.QC.report.chr2.ch11\", \n .     reportFolder = \"ChIPQCreportChr2.chr11\")",
            "2. ChIPQCreport(chipObj, reportName = \"ChIP.QC.report.chr2.ch11\", \n .     reportFolder = \"ChIPQCreportChr2.chr11\")",
            "3. ggsave(plotRegi(object, facet = facet, facetBy = facetBy, addMetaData = addMetaData), \n .     filename = file.path(reportFolder, \"GenomicFeatureEnrichment.png\"), \n .     height = 1 * length(QCsample(object)), width = 8)",
            "4. grid.draw(plot)",
            "5. plotRegi(object, facet = facet, facetBy = facetBy, addMetaData = addMetaData)",
            "6. plotRegi(object, facet = facet, facetBy = facetBy, addMetaData = addMetaData)",
            "7. data.frame(Sample = meltedDF$Var2, GenomicIntervals = meltedDF$Var1, \n .     log2_Enrichment = meltedDF$value)",
            "8. stop(gettextf(\"arguments imply differing number of rows: %s\", \n .     paste(unique(nrows), collapse = \", \")), domain = NA)"
          ]
        }
      ],
      "execution_count": 12,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "code",
      "source": [],
      "outputs": [],
      "execution_count": null,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    }
  ],
  "metadata": {
    "kernel_info": {
      "name": "ir"
    },
    "language_info": {
      "name": "R",
      "codemirror_mode": "r",
      "pygments_lexer": "r",
      "mimetype": "text/x-r-source",
      "file_extension": ".r",
      "version": "3.5.1"
    },
    "kernelspec": {
      "name": "ir",
      "language": "R",
      "display_name": "R"
    },
    "nteract": {
      "version": "0.12.3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 4
}