{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Homework 4.3"
      ],
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 1.\n",
        "Run ChIPQC for:\n",
        "https://drive.google.com/open?id=1dKLt_m_vnHjNVo-1qoACu9GZZC9OBa9v  \n",
        "(Links to an external site.)\n",
        "\n",
        "__EFF_H3K4Me1_1_srt_dupr_chr11.bam__  \n",
        "__EFF_H3K4Me1_2_srt_dupr_chr11.bam__  \n",
        "__EFF_Input_srt_dupr_chr11.bam (Control)__  "
      ],
      "metadata": {}
    },
    {
      "cell_type": "markdown",
      "source": [
        "### - Sort and index bam file  "
      ],
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cd /Users/sfchen/Documents/2018_Su/aligned_bam_homework_1\n",
        "\n",
        "printf \"EFF_H3K4Me1_1_srt_dupr_chr11\\nEFF_H3K4Me1_2_srt_dupr_chr11\\nEFF_Input_srt_dupr_chr11\\n\" > EFF.txt\n",
        "\n",
        "while read FILE; do\n",
        "\n",
        "    # Sort the bam file\n",
        "    samtools sort -o ${FILE}.srt.bam ${FILE}.bam\n",
        "\n",
        "    # Index the bam file\n",
        "    samtools index ${FILE}.srt.bam\n",
        "\ndone < EFF.txt"
      ],
      "outputs": [],
      "execution_count": 1,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### - Run MACS2  \n",
        "\n__NOTE:__ `macs2 -c/--control` The control or mock data file."
      ],
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cd /Users/sfchen/Documents/2018_Su/aligned_bam_homework_1\n",
        "\n",
        "printf \"EFF_H3K4Me1_1_srt_dupr_chr11\\nEFF_H3K4Me1_2_srt_dupr_chr11\\n\" > sample.txt\n",
        "\n",
        "while read FILE; do\n",
        "\n",
        "    # MACS2 peak calling\n",
        "    macs2 callpeak \\\n",
        "        -t ${FILE}.srt.bam \\\n",
        "        -c EFF_Input_srt_dupr_chr11.bam \\\n",
        "        -f BAM -n ${FILE}.srt \\\n",
        "        -g mm \\\n",
        "        -p 1e-2 \\\n",
        "        --broad \\\n",
        "        --nomodel \\\n",
        "        --bdg \\\n",
        "        --shift 0 \\\n",
        "        --extsize 147 \\\n",
        "        --keep-dup all &> ${FILE}.srt.MACS2.out\n",
        "\ndone < sample.txt"
      ],
      "outputs": [],
      "execution_count": 2,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### - Create sample sheet  \n",
        "\n",
        "__NOTE:__ According to [ChIPQC Manual](https://bioconductor.riken.jp/packages/3.1/bioc/manuals/ChIPQC/man/ChIPQC.pdf), provide control file path to \"bamControl\" column.  \n",
        "\n",
        "`bamReads: file path for bam file containing aligned reads for ChIP sample`  \n",
        "`bamControl: file path for bam file containing aligned reads for control sample.`"
      ],
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cat /Users/sfchen/Documents/2018_Su/aligned_bam_homework_1/ChIPQC_samplesheet_EFF.csv"
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "SampleID,Tissue,Factor,Condition,Replicate,bamReads,ControlID,bamControl,Peaks,PeakCaller\r\n",
            "H3K4Me1_1,,,,1,EFF_H3K4Me1_1_srt_dupr_chr11.srt.bam,Input,EFF_Input_srt_dupr_chr11.srt.bam,EFF_H3K4Me1_1_srt_dupr_chr11.srt_peaks.broadPeak,macs\r\n",
            "H3K4Me1_2,,,,2,EFF_H3K4Me1_2_srt_dupr_chr11.srt.bam,Input,EFF_Input_srt_dupr_chr11.srt.bam,EFF_H3K4Me1_2_srt_dupr_chr11.srt_peaks.broadPeak,macs"
          ]
        }
      ],
      "execution_count": 3,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### - ChIPQC  \n",
        "\n__NOTE:__ Check \"4.3_Homework_R.ipynb\""
      ],
      "metadata": {}
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 2. Use Homer to search for TF motifs in broadPeak file (from Q1, EFF_H3K4Me1_1_srt_dupr_chr11.bam); genome reference: genome_chr11.fa\n"
      ],
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cd /Users/sfchen/Documents/2018_Su/aligned_bam_homework_1\n",
        "bed_name=EFF_H3K4Me1_1_srt_dupr_chr11.srt_peaks.broadPeak\n",
        "chr11_genome=/Users/sfchen/Documents/2018_Su/UCSC_mm10/genome_chr11.fa\n",
        "out_name=EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF\n",
        "err_name=EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF.out\n",
        "findMotifsGenome.pl $bed_name $chr11_genome $out_name &> $err_name"
      ],
      "outputs": [],
      "execution_count": 4,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### - What are the top 5 knownResults?"
      ],
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cd /Users/sfchen/Documents/2018_Su/aligned_bam_homework_1/EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF\n",
        "head -6 knownResults.txt | cut -f 1"
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Motif Name\n",
            "ETV1(ETS)/GIST48-ETV1-ChIP-Seq(GSE22441)/Homer\n",
            "Fli1(ETS)/CD8-FLI-ChIP-Seq(GSE20898)/Homer\n",
            "ERG(ETS)/VCaP-ERG-ChIP-Seq(GSE14097)/Homer\n",
            "ETS1(ETS)/Jurkat-ETS1-ChIP-Seq(GSE17954)/Homer\n",
            "GABPA(ETS)/Jurkat-GABPa-ChIP-Seq(GSE17954)/Homer\n"
          ]
        }
      ],
      "execution_count": 7,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### - What is the difference between knownResults and homerResults?"
      ],
      "metadata": {}
    },
    {
      "cell_type": "markdown",
      "source": [
        "The top 5 best match of homerResults (aka. Homer *de novo* motif enrichment results) are:  \n",
        "\n",
        "1. eyg/dmmpmm(Bergman)/fly(0.656)  \n",
        "2. FMR1(KH)/Homo_sapiens-RNCMPT00016-PBM/HughesRNA(0.712)  \n",
        "3. TEC1(MacIsaac)/Yeast(0.707)  \n",
        "4. LIN28A(CSD)/Homo_sapiens-RNCMPT00162-PBM/HughesRNA(0.714)  \n",
        "5. FZF1/MA0298.1/Jaspar(0.667)\n",
        "\n",
        "According to the [Homer manual](http://homer.ucsd.edu/homer/introduction/basics.html):\n",
        "- *De novo* motif discovery allows to directly query the sequence to discover which motifs are the MOST enriched sequences in target set. Best saved for last.  \n",
        "- Known motif discovery will simply tell which of the known motifs is most enriched in target set. Also, it's performed first since it is usually faster, and gives a faster look at what's enriched in your target regions.\n"
      ],
      "metadata": {}
    },
    {
      "cell_type": "markdown",
      "source": [
        "__Supplement:__ stdout of Homer."
      ],
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [
        "%%bash\n",
        "cat /Users/sfchen/Documents/2018_Su/aligned_bam_homework_1/EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF.out"
      ],
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "\tPosition file = EFF_H3K4Me1_1_srt_dupr_chr11.srt_peaks.broadPeak\n",
            "\tGenome = /Users/sfchen/Documents/2018_Su/UCSC_mm10/genome_chr11.fa\n",
            "\tOutput Directory = EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF\n",
            "\tUsing Custom Genome\n",
            "\tPeak/BED file conversion summary:\n",
            "\t\tBED/Header formatted lines: 17844\n",
            "\t\tpeakfile formatted lines: 0\n",
            "\n",
            "\tPeak File Statistics:\n",
            "\t\tTotal Peaks: 17844\n",
            "\t\tRedundant Peak IDs: 0\n",
            "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
            "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
            "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
            "\n",
            "\tPeak file looks good!\n",
            "\n",
            "\tBackground files for 200 bp fragments found.\n",
            "\n",
            "\tExtracting sequences from file: /Users/sfchen/Documents/2018_Su/UCSC_mm10/genome_chr11.fa\n",
            "\tLooking for peak sequences in a single file (/Users/sfchen/Documents/2018_Su/UCSC_mm10/genome_chr11.fa)\n",
            "\tExtracting 17844 sequences from chr11\n",
            "\n",
            "\tNot removing redundant sequences\n",
            "\n\n",
            "\tSequences processed:\n",
            "\t\tAuto detected maximum sequence length of 201 bp\n",
            "\t\t17844 total\n",
            "\n",
            "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
            "\tFreq\tBin\tCount\n",
            "\t0.3\t2\t8\n",
            "\t0.35\t3\t115\n",
            "\t0.4\t4\t969\n",
            "\t0.45\t5\t3715\n",
            "\t0.5\t6\t5995\n",
            "\t0.6\t7\t6518\n",
            "\t0.7\t8\t482\n",
            "\t0.8\t9\t42\n",
            "\n",
            "\tTotal sequences set to 50000\n",
            "\n",
            "\tChoosing background that matches in CpG/GC content...\n",
            "\tBin\t# Targets\t# Background\tBackground Weight\n",
            "\t2\t8\t14\t1.030\n",
            "\t3\t115\t207\t1.001\n",
            "\t4\t969\t1746\t1.000\n",
            "\t5\t3715\t6695\t1.000\n",
            "\t6\t5995\t10803\t1.000\n",
            "\t7\t6518\t11746\t1.000\n",
            "\t8\t482\t869\t1.000\n",
            "\t9\t42\t76\t0.996\n",
            "\tAssembling sequence file...\n",
            "\tNormalizing lower order oligos using homer2\n",
            "\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tAutonormalization: 1-mers (4 total)\n",
            "\t\tA\t25.67%\t25.76%\t0.996\n",
            "\t\tC\t24.33%\t24.24%\t1.004\n",
            "\t\tG\t24.33%\t24.24%\t1.004\n",
            "\t\tT\t25.67%\t25.76%\t0.996\n",
            "\tAutonormalization: 2-mers (16 total)\n",
            "\t\tAA\t6.73%\t6.85%\t0.983\n",
            "\t\tCA\t8.13%\t7.89%\t1.030\n",
            "\t\tGA\t6.35%\t6.51%\t0.974\n",
            "\t\tTA\t4.45%\t4.50%\t0.989\n",
            "\t\tAC\t5.61%\t5.43%\t1.034\n",
            "\t\tCC\t6.74%\t6.93%\t0.973\n",
            "\t\tGC\t5.64%\t5.37%\t1.050\n",
            "\t\tTC\t6.35%\t6.51%\t0.974\n",
            "\t\tAG\t8.13%\t8.17%\t0.995\n",
            "\t\tCG\t1.34%\t1.25%\t1.069\n",
            "\t\tGG\t6.74%\t6.93%\t0.973\n",
            "\t\tTG\t8.13%\t7.89%\t1.030\n",
            "\t\tAT\t5.18%\t5.30%\t0.978\n",
            "\t\tCT\t8.13%\t8.17%\t0.995\n",
            "\t\tGT\t5.61%\t5.43%\t1.034\n",
            "\t\tTT\t6.73%\t6.85%\t0.983\n",
            "\tAutonormalization: 3-mers (64 total)\n",
            "\tNormalization weights can be found in file: EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF/seq.autonorm.tsv\n",
            "\tConverging on autonormalization solution:\n",
            "\t...............................................................................\n",
            "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
            "\t\tA\t25.67%\t25.67%\t1.000\n",
            "\t\tC\t24.33%\t24.33%\t1.000\n",
            "\t\tG\t24.33%\t24.33%\t1.000\n",
            "\t\tT\t25.67%\t25.67%\t1.000\n",
            "\tAutonormalization: 2-mers (16 total)\n",
            "\t\tAA\t6.73%\t6.74%\t1.000\n",
            "\t\tCA\t8.13%\t8.11%\t1.002\n",
            "\t\tGA\t6.35%\t6.36%\t0.997\n",
            "\t\tTA\t4.45%\t4.45%\t1.000\n",
            "\t\tAC\t5.61%\t5.59%\t1.004\n",
            "\t\tCC\t6.74%\t6.76%\t0.996\n",
            "\t\tGC\t5.64%\t5.62%\t1.004\n",
            "\t\tTC\t6.35%\t6.36%\t0.997\n",
            "\t\tAG\t8.13%\t8.13%\t1.000\n",
            "\t\tCG\t1.34%\t1.33%\t1.006\n",
            "\t\tGG\t6.74%\t6.76%\t0.996\n",
            "\t\tTG\t8.13%\t8.11%\t1.002\n",
            "\t\tAT\t5.18%\t5.20%\t0.996\n",
            "\t\tCT\t8.13%\t8.13%\t1.000\n",
            "\t\tGT\t5.61%\t5.59%\t1.004\n",
            "\t\tTT\t6.73%\t6.74%\t1.000\n",
            "\tAutonormalization: 3-mers (64 total)\n",
            "\tFinished preparing sequence/group files\n",
            "\n",
            "\t----------------------------------------------------------\n",
            "\tKnown motif enrichment\n",
            "\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\t976 motifs loaded\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\tChecking enrichment of 976 motif(s)\n",
            "\t|0%                                    50%                                  100%|\n",
            "\t=================================================================================\n",
            "\tPreparing HTML output with sequence logos...\n",
            "\t\t1 of 976 (1e-17) ETV1(ETS)/GIST48-ETV1-ChIP-Seq(GSE22441)/Homer\n",
            "\t\t2 of 976 (1e-15) Fli1(ETS)/CD8-FLI-ChIP-Seq(GSE20898)/Homer\n",
            "\t\t3 of 976 (1e-15) ERG(ETS)/VCaP-ERG-ChIP-Seq(GSE14097)/Homer\n",
            "\t\t4 of 976 (1e-15) ETS1(ETS)/Jurkat-ETS1-ChIP-Seq(GSE17954)/Homer\n",
            "\t\t5 of 976 (1e-14) GABPA(ETS)/Jurkat-GABPa-ChIP-Seq(GSE17954)/Homer\n",
            "\t\t6 of 976 (1e-13) Etv2(ETS)/ES-ER71-ChIP-Seq(GSE59402)/Homer(0.967)\n",
            "\t\t7 of 976 (1e-12) Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer\n",
            "\t\t8 of 976 (1e-11) Ets1-distal(ETS)/CD4+-PolII-ChIP-Seq(Barski_et_al.)/Homer\n",
            "\t\t9 of 976 (1e-11) RUNX-AML(Runt)/CD4+-PolII-ChIP-Seq(Barski_et_al.)/Homer\n",
            "\t\t10 of 976 (1e-9) ETV4(ETS)/HepG2-ETV4-ChIP-Seq(ENCODE)/Homer\n",
            "\t\t11 of 976 (1e-9) EWS:FLI1-fusion(ETS)/SK_N_MC-EWS:FLI1-ChIP-Seq(SRA014231)/Homer\n",
            "\t\t12 of 976 (1e-9) EHF(ETS)/LoVo-EHF-ChIP-Seq(GSE49402)/Homer\n",
            "\t\t13 of 976 (1e-8) ETS(ETS)/Promoter/Homer\n",
            "\t\t14 of 976 (1e-8) PU.1(ETS)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer\n",
            "\t\t15 of 976 (1e-8) Elk1(ETS)/Hela-Elk1-ChIP-Seq(GSE31477)/Homer\n",
            "\t\t16 of 976 (1e-7) BMYB(HTH)/Hela-BMYB-ChIP-Seq(GSE27030)/Homer\n",
            "\t\t17 of 976 (1e-7) ELF3(ETS)/PDAC-ELF3-ChIP-Seq(GSE64557)/Homer\n",
            "\t\t18 of 976 (1e-5) EWS:ERG-fusion(ETS)/CADO_ES1-EWS:ERG-ChIP-Seq(SRA014231)/Homer\n",
            "\t\t19 of 976 (1e-5) KLF6(Zf)/PDAC-KLF6-ChIP-Seq(GSE64557)/Homer\n",
            "\t\t20 of 976 (1e-5) ELF1(ETS)/Jurkat-ELF1-ChIP-Seq(SRA014231)/Homer\n",
            "\t\t21 of 976 (1e-5) Elk4(ETS)/Hela-Elk4-ChIP-Seq(GSE31477)/Homer\n",
            "\t\t22 of 976 (1e-5) GATA3(Zf)/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
            "\t\t23 of 976 (1e-5) Gata4(Zf)/Heart-Gata4-ChIP-Seq(GSE35151)/Homer\n",
            "\t\t24 of 976 (1e-5) SPDEF(ETS)/VCaP-SPDEF-ChIP-Seq(SRA014231)/Homer\n",
            "\t\t25 of 976 (1e-4) RORa(NR)/Liver-Rora-ChIP-Seq(GSE101115)/Homer\n",
            "\t\t26 of 976 (1e-4) Replumless(BLH)/Arabidopsis-RPL.GFP-ChIP-Seq(GSE78727)/Homer\n",
            "\t\t27 of 976 (1e-4) CEBP:AP1(bZIP)/ThioMac-CEBPb-ChIP-Seq(GSE21512)/Homer\n",
            "\t\t28 of 976 (1e-4) At3g60580(C2H2)/col-At3g60580-DAP-Seq(GSE60143)/Homer\n",
            "\t\t29 of 976 (1e-4) CAMTA5(CAMTA)/col-CAMTA5-DAP-Seq(GSE60143)/Homer\n",
            "\t\t30 of 976 (1e-4) KAN2(G2like)/colamp-KAN2-DAP-Seq(GSE60143)/Homer\n",
            "\t\t31 of 976 (1e-4) Tbx5(T-box)/HL1-Tbx5.biotin-ChIP-Seq(GSE21529)/Homer\n",
            "\t\t32 of 976 (1e-4) At1g64620(C2C2dof)/colamp-At1g64620-DAP-Seq(GSE60143)/Homer\n",
            "\t\t33 of 976 (1e-4) Tbr1(T-box)/Cortex-Tbr1-ChIP-Seq(GSE71384)/Homer\n",
            "\t\t34 of 976 (1e-4) RUNX2(Runt)/PCa-RUNX2-ChIP-Seq(GSE33889)/Homer\n",
            "\t\t35 of 976 (1e-4) MYB118(MYB)/colamp-MYB118-DAP-Seq(GSE60143)/Homer\n",
            "\t\t36 of 976 (1e-4) bZIP50(bZIP)/colamp-bZIP50-DAP-Seq(GSE60143)/Homer\n",
            "\t\t37 of 976 (1e-4) ELF5(ETS)/T47D-ELF5-ChIP-Seq(GSE30407)/Homer\n",
            "\t\t38 of 976 (1e-4) TGA2(bZIP)/colamp-TGA2-DAP-Seq(GSE60143)/Homer\n",
            "\t\t39 of 976 (1e-4) SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer\n",
            "\t\t40 of 976 (1e-4) RORg(NR)/Liver-Rorc-ChIP-Seq(GSE101115)/Homer\n",
            "\t\t41 of 976 (1e-4) MYB62(MYB)/colamp-MYB62-DAP-Seq(GSE60143)/Homer\n",
            "\t\t42 of 976 (1e-4) EBF2(EBF)/BrownAdipose-EBF2-ChIP-Seq(GSE97114)/Homer\n",
            "\t\t43 of 976 (1e-3) AT3G52440(C2C2dof)/colamp-AT3G52440-DAP-Seq(GSE60143)/Homer\n",
            "\t\t44 of 976 (1e-3) Tbx6(T-box)/ESC-Tbx6-ChIP-Seq(GSE93524)/Homer\n",
            "\t\t45 of 976 (1e-3) Sp2(Zf)/HEK293-Sp2.eGFP-ChIP-Seq(Encode)/Homer\n",
            "\t\t46 of 976 (1e-3) RUNX1(Runt)/Jurkat-RUNX1-ChIP-Seq(GSE29180)/Homer\n",
            "\t\t47 of 976 (1e-3) AT2G15740(C2H2)/col-AT2G15740-DAP-Seq(GSE60143)/Homer\n",
            "\t\t48 of 976 (1e-3) MYB70(MYB)/col-MYB70-DAP-Seq(GSE60143)/Homer\n",
            "\t\t49 of 976 (1e-3) AT2G28810(C2C2dof)/colamp-AT2G28810-DAP-Seq(GSE60143)/Homer\n",
            "\t\t50 of 976 (1e-3) dof42(C2C2dof)/col-dof42-DAP-Seq(GSE60143)/Homer\n",
            "\t\t51 of 976 (1e-3) Gata6(Zf)/HUG1N-GATA6-ChIP-Seq(GSE51936)/Homer\n",
            "\t\t52 of 976 (1e-3) SF1(NR)/H295R-Nr5a1-ChIP-Seq(GSE44220)/Homer\n",
            "\t\t53 of 976 (1e-3) At5g04390(C2H2)/col200-At5g04390-DAP-Seq(GSE60143)/Homer\n",
            "\t\t54 of 976 (1e-3) DMRT1(DM)/Testis-DMRT1-ChIP-Seq(GSE64892)/Homer\n",
            "\t\t55 of 976 (1e-3) BOS1(MYB)/col-BOS1-DAP-Seq(GSE60143)/Homer\n",
            "\t\t56 of 976 (1e-3) MYB81(MYB)/col-MYB81-DAP-Seq(GSE60143)/Homer\n",
            "\t\t57 of 976 (1e-3) TGA6(bZIP)/colamp-TGA6-DAP-Seq(GSE60143)/Homer\n",
            "\t\t58 of 976 (1e-3) CRX(Homeobox)/Retina-Crx-ChIP-Seq(GSE20012)/Homer\n",
            "\t\t59 of 976 (1e-3) RUNX(Runt)/HPC7-Runx1-ChIP-Seq(GSE22178)/Homer\n",
            "\t\t60 of 976 (1e-3) Tgif2(Homeobox)/mES-Tgif2-ChIP-Seq(GSE55404)/Homer\n",
            "\t\t61 of 976 (1e-3) ANAC045(NAC)/col-ANAC045-DAP-Seq(GSE60143)/Homer\n",
            "\t\t62 of 976 (1e-3) GBF6(bZIP)/colamp-GBF6-DAP-Seq(GSE60143)/Homer\n",
            "\t\t63 of 976 (1e-3) MYB3R5(MYB)/col-MYB3R5-DAP-Seq(GSE60143)/Homer\n",
            "\t\t64 of 976 (1e-3) EKLF(Zf)/Erythrocyte-Klf1-ChIP-Seq(GSE20478)/Homer\n",
            "\t\t65 of 976 (1e-3) dof24(C2C2dof)/col-dof24-DAP-Seq(GSE60143)/Homer\n",
            "\t\t66 of 976 (1e-3) Nur77(NR)/K562-NR4A1-ChIP-Seq(GSE31363)/Homer\n",
            "\t\t67 of 976 (1e-3) PBX1(Homeobox)/MCF7-PBX1-ChIP-Seq(GSE28007)/Homer\n",
            "\t\t68 of 976 (1e-3) JGL(C2H2)/col-JGL-DAP-Seq(GSE60143)/Homer\n",
            "\t\t69 of 976 (1e-3) HDG7(HB)/col-HDG7-DAP-Seq(GSE60143)/Homer\n",
            "\t\t70 of 976 (1e-3) At4g32800(AP2EREBP)/colamp-At4g32800-DAP-Seq(GSE60143)/Homer\n",
            "\t\t71 of 976 (1e-3) NFAT:AP1(RHD,bZIP)/Jurkat-NFATC1-ChIP-Seq(Jolma_et_al.)/Homer\n",
            "\t\t72 of 976 (1e-2) DMRT6(DM)/Testis-DMRT6-ChIP-Seq(GSE60440)/Homer\n",
            "\t\t73 of 976 (1e-2) AZF1(C2H2)/colamp-AZF1-DAP-Seq(GSE60143)/Homer\n",
            "\t\t74 of 976 (1e-2) CDF3(C2C2dof)/colamp-CDF3-DAP-Seq(GSE60143)/Homer\n",
            "\t\t75 of 976 (1e-2) TGA4(bZIP)/colamp-TGA4-DAP-Seq(GSE60143)/Homer\n",
            "\t\t76 of 976 (1e-2) AREB3(bZIP)/col-AREB3-DAP-Seq(GSE60143)/Homer\n",
            "\t\t77 of 976 (1e-2) AT1G69570(C2C2dof)/col-AT1G69570-DAP-Seq(GSE60143)/Homer\n",
            "\t\t78 of 976 (1e-2) TGA5(bZIP)/col-TGA5-DAP-Seq(GSE60143)/Homer\n",
            "\t\t79 of 976 (1e-2) ZNF675(Zf)/HEK293-ZNF675.GFP-ChIP-Seq(GSE58341)/Homer\n",
            "\t\t80 of 976 (1e-2) AT5G66940(C2C2dof)/col-AT5G66940-DAP-Seq(GSE60143)/Homer\n",
            "\t\t81 of 976 (1e-2) bHLH157(bHLH)/col-bHLH157-DAP-Seq(GSE60143)/Homer\n",
            "\t\t82 of 976 (1e-2) Knotted(Homeobox)/Corn-KN1-ChIP-Seq(GSE39161)/Homer\n",
            "\t\t83 of 976 (1e-2) GBF5(bZIP)/colamp-GBF5-DAP-Seq(GSE60143)/Homer\n",
            "\t\t84 of 976 (1e-2) AT3G12130(C3H)/colamp-AT3G12130-DAP-Seq(GSE60143)/Homer\n",
            "\t\t85 of 976 (1e-2) Foxh1(Forkhead)/hESC-FOXH1-ChIP-Seq(GSE29422)/Homer\n",
            "\t\t86 of 976 (1e-2) Nr5a2(NR)/mES-Nr5a2-ChIP-Seq(GSE19019)/Homer\n",
            "\t\t87 of 976 (1e-2) MYB3R4(MYB)/col-MYB3R4-DAP-Seq(GSE60143)/Homer\n",
            "\t\t88 of 976 (1e-2) MYB65(MYB)/colamp-MYB65-DAP-Seq(GSE60143)/Homer\n",
            "\t\t89 of 976 (1e-2) THRa(NR)/C17.2-THRa-ChIP-Seq(GSE38347)/Homer\n",
            "\t\t90 of 976 (1e-2) Unknown-ESC-element(?)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
            "\t\t91 of 976 (1e-2) Gata1(Zf)/K562-GATA1-ChIP-Seq(GSE18829)/Homer\n",
            "\t\t92 of 976 (1e-2) DAG2(C2C2dof)/col-DAG2-DAP-Seq(GSE60143)/Homer\n",
            "\t\t93 of 976 (1e-2) ETS:RUNX(ETS,Runt)/Jurkat-RUNX1-ChIP-Seq(GSE17954)/Homer\n",
            "\t\t94 of 976 (1e-2) TGA1(bZIP)/colamp-TGA1-DAP-Seq(GSE60143)/Homer\n",
            "\t\t95 of 976 (1e-2) KLF5(Zf)/LoVo-KLF5-ChIP-Seq(GSE49402)/Homer\n",
            "\t\t96 of 976 (1e-2) AT5G63260(C3H)/col-AT5G63260-DAP-Seq(GSE60143)/Homer\n",
            "\t\t97 of 976 (1e-2) MYB116(MYB)/colamp-MYB116-DAP-Seq(GSE60143)/Homer\n",
            "\t\t98 of 976 (1e-2) VND6(NAC)/col-VND6-DAP-Seq(GSE60143)/Homer\n",
            "\t\t99 of 976 (1e-2) E2F6(E2F)/Hela-E2F6-ChIP-Seq(GSE31477)/Homer\n",
            "\t\t100 of 976 (1e-2) DREF/Drosophila-Promoters/Homer\n",
            "\t\t101 of 976 (1e-2) Atf4(bZIP)/MEF-Atf4-ChIP-Seq(GSE35681)/Homer\n",
            "\t\t102 of 976 (1e-2) Pitx1(Homeobox)/Chicken-Pitx1-ChIP-Seq(GSE38910)/Homer\n",
            "\t\t103 of 976 (1e-2) THRb(NR)/HepG2-THRb.Flag-ChIP-Seq(Encode)/Homer\n",
            "\t\t104 of 976 (1e-2) TGA9(bZIP)/colamp-TGA9-DAP-Seq(GSE60143)/Homer\n",
            "\t\t105 of 976 (1e-2) Bach2(bZIP)/OCILy7-Bach2-ChIP-Seq(GSE44420)/Homer\n",
            "\t\t106 of 976 (1e-2) Chop(bZIP)/MEF-Chop-ChIP-Seq(GSE35681)/Homer\n",
            "\t\t107 of 976 (1e-2) At1g49010(MYBrelated)/col-At1g49010-DAP-Seq(GSE60143)/Homer\n",
            "\t\t108 of 976 (1e-2) AMYB(HTH)/Testes-AMYB-ChIP-Seq(GSE44588)/Homer\n",
            "\t\t109 of 976 (1e-2) SPL5(SBP)/colamp-SPL5-DAP-Seq(GSE60143)/Homer\n",
            "\t\t110 of 976 (1e-2) bHLH18(bHLH)/col-bHLH18-DAP-Seq(GSE60143)/Homer\n",
            "\t\t111 of 976 (1e-2) bZIP28(bZIP)/col-bZIP28-DAP-Seq(GSE60143)/Homer\n",
            "\t\t112 of 976 (1e-2) Unknown3/Arabidopsis-Promoters/Homer\n",
            "\t\t113 of 976 (1e-2) AT5G56840(MYBrelated)/colamp-AT5G56840-DAP-Seq(GSE60143)/Homer\n",
            "\t\t114 of 976 (1e-2) DUX(Homeobox)/C2C12-Dux-ChIP-Seq(GSE87279)/Homer\n",
            "\t\t115 of 976 (1e-2) AT1G76870(Trihelix)/col-AT1G76870-DAP-Seq(GSE60143)/Homer\n",
            "\t\t116 of 976 (1e-2) ABI5(bZIP)/col-ABI5-DAP-Seq(GSE60143)/Homer\n",
            "\t\t117 of 976 (1e-2) IRF1(IRF)/PBMC-IRF1-ChIP-Seq(GSE43036)/Homer\n",
            "\t\t118 of 976 (1e-2) AT2G31460(REMB3)/col-AT2G31460-DAP-Seq(GSE60143)/Homer\n",
            "\t\t119 of 976 (1e-2) SMB(NAC)/colamp-SMB-DAP-Seq(GSE60143)/Homer\n",
            "\t\t120 of 976 (1e-2) Gata2(Zf)/K562-GATA2-ChIP-Seq(GSE18829)/Homer\n",
            "\t\t121 of 976 (1e-2) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\n",
            "\t\t122 of 976 (1e-2) TOD6?/SacCer-Promoters/Homer\n",
            "\t\t123 of 976 (1e-2) LBD23(LOBAS2)/colamp-LBD23-DAP-Seq(GSE60143)/Homer\n",
            "\t\t124 of 976 (1e-2) NLP7(RWPRK)/col-NLP7-DAP-Seq(GSE60143)/Homer\n",
            "\t\t125 of 976 (1e-2) AT5G61620(MYBrelated)/colamp-AT5G61620-DAP-Seq(GSE60143)/Homer\n",
            "\t\t126 of 976 (1e-2) bZIP53(bZIP)/colamp-bZIP53-DAP-Seq(GSE60143)/Homer\n",
            "\t\t127 of 976 (1e-2) ZML2(C2C2gata)/col-ZML2-DAP-Seq(GSE60143)/Homer\n",
            "\t\t128 of 976 (1e-2) MYB119(MYB)/colamp-MYB119-DAP-Seq(GSE60143)/Homer\n",
            "\t\t129 of 976 (1e-2) AT3G10030(Trihelix)/colamp-AT3G10030-DAP-Seq(GSE60143)/Homer\n",
            "\t\t130 of 976 (1e-2) PQM-1(?)/cElegans-L3-ChIP-Seq(modEncode)/Homer\n",
            "\t\t131 of 976 (1e-2) ZNF382(Zf)/HEK293-ZNF382.GFP-ChIP-Seq(GSE58341)/Homer\n",
            "\t\t132 of 976 (1e-2) At5g47390(MYBrelated)/col-At5g47390-DAP-Seq(GSE60143)/Homer\n",
            "\t\t133 of 976 (1e-2) KANADI1(Myb)/Seedling-KAN1-ChIP-Seq(GSE48081)/Homer\n",
            "\t\t134 of 976 (1e-2) Nkx2.2(Homeobox)/NPC-Nkx2.2-ChIP-Seq(GSE61673)/Homer\n",
            "\t\t135 of 976 (1e-2) At5g05790(MYBrelated)/col-At5g05790-DAP-Seq(GSE60143)/Homer\n",
            "\t\t136 of 976 (1e-2) ANAC053(NAC)/colamp-ANAC053-DAP-Seq(GSE60143)/Homer\n",
            "\t\t137 of 976 (1e-2) SeqBias: polyC-repeat\n",
            "\t\t138 of 976 (1e-2) Tbox:Smad(T-box,MAD)/ESCd5-Smad2_3-ChIP-Seq(GSE29422)/Homer\n",
            "\t\t139 of 976 (1e-2) NAC2(NAC)/colamp-NAC2-DAP-Seq(GSE60143)/Homer\n",
            "\t----------------------------------------------------------\n",
            "\tDe novo motif finding (HOMER)\n",
            "\n",
            "\tScanning input files...\n",
            "\tParsing sequences...\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t===============================================================================\n",
            "\tTotal number of Oligos: 32895\n",
            "\tAutoadjustment for sequence coverage in background: 1.00x\n",
            "\n",
            "\tOligos: 32895 of 34497 max\n",
            "\tTree  : 67136 of 172485 max\n",
            "\tOptimizing memory usage...\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\n",
            "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
            "\n",
            "\tScreening oligos 32895 (allowing 0 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t0.08% skipped, 99.92% checked (32868 of 32895), of those checked:\n",
            "\t\t0.08% not in target, 0.00% increased p-value, 0.00% high p-value\n",
            "\n",
            "\tScreening oligos 32895 (allowing 1 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t0.08% skipped, 99.92% checked (32868 of 32895), of those checked:\n",
            "\t\t0.00% not in target, 45.60% increased p-value, 41.28% high p-value\n",
            "\n",
            "\tScreening oligos 32895 (allowing 2 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t63.86% skipped, 36.14% checked (11887 of 32895), of those checked:\n",
            "\t\t0.00% not in target, 25.30% increased p-value, 0.00% high p-value\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\n",
            "\tLocal Optimization Phase:\n",
            "\t1 of 25 Initial Sequence: CGGTTTAA... (-29.556)\n",
            "\t\tRound 1: -48.92 CGGTTTAA T:8572.0(38.15%),B:13734.4(34.71%),P:1e-21\n",
            "\t\tRound 2: -55.35 CGGTTTAA T:8239.0(36.98%),B:13066.3(33.34%),P:1e-24\n",
            "\t\tRound 3: -55.35 CGGTTTAA T:8239.0(36.98%),B:13066.3(33.34%),P:1e-24\n",
            "\t\t=Final=: -47.28 CGGTTTAA T:7287.0(40.84%),B:12053.3(37.41%),P:1e-20\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t2 of 25 Initial Sequence: AACCCTAA... (-31.187)\n",
            "\t\tRound 1: -39.43 AACCCTAA T:315.0(1.75%),B:336.2(1.04%),P:1e-17\n",
            "\t\tRound 2: -41.14 AACGCTAA T:389.0(2.16%),B:433.8(1.34%),P:1e-17\n",
            "\t\tRound 3: -41.14 AACGCTAA T:389.0(2.16%),B:433.8(1.34%),P:1e-17\n",
            "\t\t=Final=: -21.76 AACGCTAA T:333.0(1.87%),B:421.9(1.31%),P:1e-9\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t3 of 25 Initial Sequence: CCAAAGCA... (-28.958)\n",
            "\t\tRound 1: -47.47 CCAAAGCA T:8076.0(36.40%),B:12929.3(33.06%),P:1e-20\n",
            "\t\tRound 2: -47.47 CCAAAGCA T:8076.0(36.40%),B:12929.3(33.06%),P:1e-20\n",
            "\t\t=Final=: -44.82 CCAAAGCA T:8170.0(45.79%),B:13658.3(42.40%),P:1e-19\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t4 of 25 Initial Sequence: AGATTACG... (-27.545)\n",
            "\t\tRound 1: -49.29 AGATTACG T:2661.0(13.85%),B:3934.5(11.50%),P:1e-21\n",
            "\t\tRound 2: -49.29 AGATTACG T:2661.0(13.85%),B:3934.5(11.50%),P:1e-21\n",
            "\t\t=Final=: -51.74 AGATTACG T:2479.0(13.89%),B:3696.5(11.47%),P:1e-22\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t5 of 25 Initial Sequence: CGCAGTGT... (-23.543)\n",
            "\t\tRound 1: -39.48 CGCAGTGT T:8083.0(36.43%),B:13090.4(33.39%),P:1e-17\n",
            "\t\tRound 2: -43.29 CGCAGTGT T:10153.0(43.39%),B:16502.5(40.08%),P:1e-18\n",
            "\t\tRound 3: -43.29 CGCAGTGT T:10153.0(43.39%),B:16502.5(40.08%),P:1e-18\n",
            "\t\t=Final=: -36.89 CGCAGTGT T:3889.0(21.79%),B:6225.5(19.32%),P:1e-16\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t6 of 25 Initial Sequence: TTCGGCGC... (-21.517)\n",
            "\t\tRound 1: -21.52 TTCGGCGC T:11.0(0.06%),B:1.8(0.00%),P:1e-9\n",
            "\t\tRound 2: -21.52 TTCGGCGC T:11.0(0.06%),B:1.8(0.00%),P:1e-9\n",
            "\t\t=Final=: -24.51 TTCGGCGC T:11.0(0.06%),B:1.8(0.01%),P:1e-10\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t7 of 25 Initial Sequence: GTCCAACG... (-20.026)\n",
            "\t\tRound 1: -21.48 GTCCAACG T:21.0(0.12%),B:7.5(0.02%),P:1e-9\n",
            "\t\tRound 2: -21.48 GTCCAACG T:21.0(0.12%),B:7.5(0.02%),P:1e-9\n",
            "\t\t=Final=: -20.62 GTCCAACG T:21.0(0.12%),B:7.5(0.02%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t8 of 25 Initial Sequence: TTTTGAAC... (-19.372)\n",
            "\t\tRound 1: -19.47 TTTTGAAC T:205.0(1.14%),B:238.1(0.74%),P:1e-8\n",
            "\t\tRound 2: -19.47 TTTTGAAC T:205.0(1.14%),B:238.1(0.74%),P:1e-8\n",
            "\t\t=Final=: -18.28 TTTTGAAC T:201.0(1.13%),B:238.1(0.74%),P:1e-7\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t9 of 25 Initial Sequence: AGAACGGT... (-19.040)\n",
            "\t\tRound 1: -31.76 AGAACGGT T:2135.0(11.28%),B:3238.9(9.56%),P:1e-13\n",
            "\t\tRound 2: -31.76 AGAACGGT T:2135.0(11.28%),B:3238.9(9.56%),P:1e-13\n",
            "\t\t=Final=: -30.11 AGAACGGT T:1830.0(10.26%),B:2790.1(8.66%),P:1e-13\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t10 of 25 Initial Sequence: CGTTAAAT... (-18.618)\n",
            "\t\tRound 1: -18.62 CGTTAAAT T:10.0(0.06%),B:2.0(0.01%),P:1e-8\n",
            "\t\tRound 2: -18.62 CGTTAAAT T:10.0(0.06%),B:2.0(0.01%),P:1e-8\n",
            "\t\t=Final=: -15.09 CGTTAAAT T:10.0(0.06%),B:2.0(0.01%),P:1e-6\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t11 of 25 Initial Sequence: CGCTTCCT... (-18.177)\n",
            "\t\tRound 1: -30.00 CGCTTCCT T:2366.0(12.42%),B:3636.7(10.67%),P:1e-13\n",
            "\t\tRound 2: -30.00 CGCTTCCT T:2366.0(12.42%),B:3636.7(10.67%),P:1e-13\n",
            "\t\t=Final=: -33.33 CGCTTCCT T:2201.0(12.33%),B:3383.1(10.50%),P:1e-14\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t12 of 25 Initial Sequence: GTAGCATT... (-16.963)\n",
            "\t\tRound 1: -23.54 GTAGCATT T:328.0(1.82%),B:405.3(1.25%),P:1e-10\n",
            "\t\tRound 2: -25.40 GTAGCACT T:848.0(4.64%),B:1201.2(3.66%),P:1e-11\n",
            "\t\tRound 3: -27.89 GTAGCATT T:520.0(2.87%),B:675.8(2.07%),P:1e-12\n",
            "\t\tRound 4: -27.89 GTAGCATT T:520.0(2.87%),B:675.8(2.07%),P:1e-12\n",
            "\t\t=Final=: -29.27 GTAGCATT T:516.0(2.89%),B:668.0(2.07%),P:1e-12\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t13 of 25 Initial Sequence: TTACACGC... (-16.682)\n",
            "\t\tRound 1: -31.05 TTACACGC T:533.0(2.94%),B:681.7(2.09%),P:1e-13\n",
            "\t\tRound 2: -31.05 TTACACGC T:533.0(2.94%),B:681.7(2.09%),P:1e-13\n",
            "\t\t=Final=: -26.79 TTACACGC T:512.0(2.87%),B:672.8(2.09%),P:1e-11\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t14 of 25 Initial Sequence: GATCGTAC... (-15.823)\n",
            "\t\tRound 1: -16.06 YATCGTAH T:881.0(4.82%),B:1324.5(4.03%),P:1e-6\n",
            "\t\tRound 2: -16.06 YATCGTAH T:881.0(4.82%),B:1324.5(4.03%),P:1e-6\n",
            "\t\t=Final=: -18.53 YATCGTAH T:861.0(4.83%),B:1280.3(3.97%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t15 of 25 Initial Sequence: GTAACACG... (-16.384)\n",
            "\t\tRound 1: -18.46 GTAACACG T:1600.0(8.58%),B:2492.8(7.44%),P:1e-8\n",
            "\t\tRound 2: -21.29 DTAAYWCG T:5215.0(25.34%),B:8585.7(23.39%),P:1e-9\n",
            "\t\tRound 3: -24.90 DWRACDCG T:6705.0(31.32%),B:11054.3(29.05%),P:1e-10\n",
            "\t\tRound 4: -24.90 DWRACDCG T:6705.0(31.32%),B:11054.3(29.05%),P:1e-10\n",
            "\t\t=Final=: -21.13 DWRACDCG T:5206.0(29.18%),B:8743.5(27.14%),P:1e-9\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t16 of 25 Initial Sequence: TACGTACG... (-15.823)\n",
            "\t\tRound 1: -20.46 TAKGTGCG T:76.0(0.43%),B:63.4(0.20%),P:1e-8\n",
            "\t\tRound 2: -21.65 TATGTGCG T:68.0(0.38%),B:52.3(0.16%),P:1e-9\n",
            "\t\tRound 3: -21.65 TATGTGCG T:68.0(0.38%),B:52.3(0.16%),P:1e-9\n",
            "\t\t=Final=: -17.55 TATGTGCG T:63.0(0.35%),B:52.3(0.16%),P:1e-7\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t17 of 25 Initial Sequence: GCGCTTCG... (-15.423)\n",
            "\t\tRound 1: -24.60 GCGCTTCG T:21.0(0.12%),B:6.7(0.02%),P:1e-10\n",
            "\t\tRound 2: -24.60 GCGCTTCG T:21.0(0.12%),B:6.7(0.02%),P:1e-10\n",
            "\t\t=Final=: -23.33 GCGCTTCG T:21.0(0.12%),B:6.7(0.02%),P:1e-10\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t18 of 25 Initial Sequence: TTAGTCGA... (-15.423)\n",
            "\t\tRound 1: -24.54 TTAGTCGA T:44.0(0.25%),B:24.2(0.07%),P:1e-10\n",
            "\t\tRound 2: -24.54 TTAGTCGA T:44.0(0.25%),B:24.2(0.07%),P:1e-10\n",
            "\t\t=Final=: -24.45 TTAGTCGA T:44.0(0.25%),B:24.2(0.08%),P:1e-10\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t19 of 25 Initial Sequence: TACGTCAA... (-15.086)\n",
            "\t\tRound 1: -20.17 TACGTCAA T:34.0(0.19%),B:18.2(0.06%),P:1e-8\n",
            "\t\tRound 2: -20.17 TACGTCAA T:34.0(0.19%),B:18.2(0.06%),P:1e-8\n",
            "\t\t=Final=: -20.05 TACGTCAA T:34.0(0.19%),B:18.2(0.06%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t20 of 25 Initial Sequence: GCGGAGAG... (-15.063)\n",
            "\t\tRound 1: -21.55 GCGGAGAG T:160.0(0.89%),B:170.4(0.53%),P:1e-9\n",
            "\t\tRound 2: -22.78 GCGGAAAG T:357.0(1.98%),B:451.2(1.39%),P:1e-9\n",
            "\t\tRound 3: -22.78 GCGGAAAG T:357.0(1.98%),B:451.2(1.39%),P:1e-9\n",
            "\t\t=Final=: -21.79 GCGGAAAG T:345.0(1.93%),B:439.8(1.37%),P:1e-9\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t21 of 25 Initial Sequence: GGCCTGGG... (-14.677)\n",
            "\t\tRound 1: -21.81 GGCCTGGG T:1079.0(5.87%),B:1600.3(4.85%),P:1e-9\n",
            "\t\tRound 2: -21.81 GGCCTGGG T:1079.0(5.87%),B:1600.3(4.85%),P:1e-9\n",
            "\t\t=Final=: -18.99 GGCCTGGG T:1013.0(5.68%),B:1527.1(4.74%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t22 of 25 Initial Sequence: GAAACCAC... (-14.552)\n",
            "\t\tRound 1: -26.65 GAAACCAC T:501.0(2.77%),B:652.0(2.00%),P:1e-11\n",
            "\t\tRound 2: -26.65 GAAACCAC T:501.0(2.77%),B:652.0(2.00%),P:1e-11\n",
            "\t\t=Final=: -25.78 GAAACCAC T:494.0(2.77%),B:649.4(2.02%),P:1e-11\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t23 of 25 Initial Sequence: AGTAGCGA... (-14.279)\n",
            "\t\tRound 1: -22.21 AGTAGCGA T:2347.0(12.32%),B:3698.7(10.84%),P:1e-9\n",
            "\t\tRound 2: -22.21 AGTAGCGA T:2347.0(12.32%),B:3698.7(10.84%),P:1e-9\n",
            "\t\t=Final=: -20.53 AGTAGCGA T:1203.0(6.74%),B:1829.4(5.68%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t24 of 25 Initial Sequence: AAACGGAT... (-14.111)\n",
            "\t\tRound 1: -23.25 WACCGGAT T:68.0(0.38%),B:50.1(0.16%),P:1e-10\n",
            "\t\tRound 2: -25.59 TAACGGAT T:47.0(0.26%),B:26.0(0.08%),P:1e-11\n",
            "\t\tRound 3: -25.59 TAACGGAT T:47.0(0.26%),B:26.0(0.08%),P:1e-11\n",
            "\t\t=Final=: -25.52 TAACGGAT T:47.0(0.26%),B:26.0(0.08%),P:1e-11\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t25 of 25 Initial Sequence: CGTAGCAA... (-13.775)\n",
            "\t\tRound 1: -17.56 CGTAGCAA T:34.0(0.19%),B:20.5(0.06%),P:1e-7\n",
            "\t\tRound 2: -17.56 CGTAGCAA T:34.0(0.19%),B:20.5(0.06%),P:1e-7\n",
            "\t\t=Final=: -17.53 CGTAGCAA T:34.0(0.19%),B:20.5(0.06%),P:1e-7\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\n",
            "\tFinalizing Enrichment Statistics (new in v3.4)\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\tChecking enrichment of 25 motif(s)\n",
            "\t|0%                                    50%                                  100%|\n",
            "\t=================================================================================\n",
            "\tOutput in file: EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF/homerMotifs.motifs8\n",
            "\n\n",
            "\tScanning input files...\n",
            "\tParsing sequences...\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t===============================================================================\n",
            "\tTotal number of Oligos: 492887\n",
            "\tAutoadjustment for sequence coverage in background: 1.00x\n",
            "\n",
            "\tOligos: 492887 of 551887 max\n",
            "\tTree  : 1018548 of 2759435 max\n",
            "\tOptimizing memory usage...\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\n",
            "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
            "\n",
            "\tScreening oligos 492887 (allowing 0 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t10.19% skipped, 89.81% checked (442640 of 492887), of those checked:\n",
            "\t\t10.19% not in target, 0.00% increased p-value, 0.00% high p-value\n",
            "\n",
            "\tScreening oligos 492887 (allowing 1 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t===============================================================================\n",
            "\t\t10.19% skipped, 89.81% checked (442640 of 492887), of those checked:\n",
            "\t\t0.00% not in target, 40.87% increased p-value, 44.67% high p-value\n",
            "\n",
            "\tScreening oligos 492887 (allowing 2 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t===============================================================================\n",
            "\t\t71.59% skipped, 28.41% checked (140007 of 492887), of those checked:\n",
            "\t\t0.00% not in target, 18.32% increased p-value, 0.00% high p-value\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\n",
            "\tLocal Optimization Phase:\n",
            "\t1 of 25 Initial Sequence: GGGTTAGGGT... (-80.373)\n",
            "\t\tRound 1: -82.42 GGGTTAGGGT T:69.0(0.39%),B:17.9(0.05%),P:1e-35\n",
            "\t\tRound 2: -93.11 GCGTTAGGGT T:76.0(0.43%),B:18.7(0.06%),P:1e-40\n",
            "\t\tRound 3: -93.11 GCGTTAGGGT T:76.0(0.43%),B:18.7(0.06%),P:1e-40\n",
            "\t\t=Final=: -12.02 GCGTTAGGGT T:27.0(0.15%),B:18.7(0.06%),P:1e-5\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t2 of 25 Initial Sequence: CAAACCAAAC... (-33.513)\n",
            "\t\tRound 1: -47.26 CAAACCAAAC T:290.0(1.61%),B:284.1(0.88%),P:1e-20\n",
            "\t\tRound 2: -47.26 CAAACCAAAC T:290.0(1.61%),B:284.1(0.88%),P:1e-20\n",
            "\t\t=Final=: -22.02 CAAACCAAAC T:246.0(1.38%),B:292.0(0.91%),P:1e-9\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t3 of 25 Initial Sequence: GGATAGACAC... (-33.387)\n",
            "\t\tRound 1: -40.67 GGATAGACAC T:2608.0(13.60%),B:3926.6(11.47%),P:1e-17\n",
            "\t\tRound 2: -40.78 GGATAGWCAC T:2793.0(14.49%),B:4228.7(12.30%),P:1e-17\n",
            "\t\tRound 3: -44.98 GGATAGWCAC T:2812.0(14.58%),B:4218.4(12.27%),P:1e-19\n",
            "\t\tRound 4: -54.97 GGATAGWCAC T:2298.0(12.08%),B:3304.8(9.75%),P:1e-23\n",
            "\t\tRound 5: -54.97 GGATAGWCAC T:2298.0(12.08%),B:3304.8(9.75%),P:1e-23\n",
            "\t\t=Final=: -58.17 GGATAGWCAC T:2150.0(12.05%),B:3109.5(9.65%),P:1e-25\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t4 of 25 Initial Sequence: CAGGAAGTGT... (-33.653)\n",
            "\t\tRound 1: -46.48 CAGGAAGTGT T:2308.0(12.13%),B:3387.8(9.98%),P:1e-20\n",
            "\t\tRound 2: -46.48 CAGGAAGTGT T:2308.0(12.13%),B:3387.8(9.98%),P:1e-20\n",
            "\t\t=Final=: -43.51 CAGGAAGTGT T:1554.0(8.71%),B:2234.2(6.94%),P:1e-18\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t5 of 25 Initial Sequence: TGCCACTTTA... (-28.939)\n",
            "\t\tRound 1: -38.81 TGCCACTTTA T:1710.0(9.14%),B:2484.6(7.42%),P:1e-16\n",
            "\t\tRound 2: -38.81 TGCCACTTTA T:1710.0(9.14%),B:2484.6(7.42%),P:1e-16\n",
            "\t\t=Final=: -42.62 TGCCACTTTA T:1626.0(9.11%),B:2357.6(7.32%),P:1e-18\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t6 of 25 Initial Sequence: GTCGTGCCCT... (-29.135)\n",
            "\t\tRound 1: -42.91 GTCGTGCCCT T:1739.0(9.29%),B:2499.3(7.46%),P:1e-18\n",
            "\t\tRound 2: -42.91 GTCGTGCCCT T:1739.0(9.29%),B:2499.3(7.46%),P:1e-18\n",
            "\t\t=Final=: -41.64 GTCGTGCCCT T:1606.0(9.00%),B:2331.6(7.24%),P:1e-18\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t7 of 25 Initial Sequence: ACACTGCGTC... (-28.096)\n",
            "\t\tRound 1: -41.13 ACACTGCGTC T:1745.0(9.32%),B:2523.4(7.53%),P:1e-17\n",
            "\t\tRound 2: -41.13 ACACTGCGTC T:1745.0(9.32%),B:2523.4(7.53%),P:1e-17\n",
            "\t\t=Final=: -34.80 ACACTGCGTC T:1516.0(8.50%),B:2232.3(6.93%),P:1e-15\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t8 of 25 Initial Sequence: CTGCTGCTCG... (-27.591)\n",
            "\t\tRound 1: -44.08 CTGCTGCTCG T:18.0(0.10%),B:1.9(0.00%),P:1e-19\n",
            "\t\tRound 2: -44.08 CTGCTGCTCG T:18.0(0.10%),B:1.9(0.00%),P:1e-19\n",
            "\t\t=Final=: -47.56 CTGCTGCTCG T:18.0(0.10%),B:1.9(0.01%),P:1e-20\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t9 of 25 Initial Sequence: ACGAGAAGAG... (-27.591)\n",
            "\t\tRound 1: -33.98 ACGAGAAGAG T:15.0(0.08%),B:2.3(0.01%),P:1e-14\n",
            "\t\tRound 2: -44.08 ACGAGAAGCG T:18.0(0.10%),B:2.3(0.01%),P:1e-19\n",
            "\t\tRound 3: -44.08 ACGAGAAGCG T:18.0(0.10%),B:2.3(0.01%),P:1e-19\n",
            "\t\t=Final=: -35.61 ACGAGAAGCG T:18.0(0.10%),B:2.3(0.01%),P:1e-15\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t10 of 25 Initial Sequence: GATCTGATAT... (-27.591)\n",
            "\t\tRound 1: -44.20 GATCTGATAT T:25.0(0.14%),B:4.7(0.01%),P:1e-19\n",
            "\t\tRound 2: -44.20 GATCTGATAT T:25.0(0.14%),B:4.7(0.01%),P:1e-19\n",
            "\t\t=Final=: -40.26 GATCTGATAT T:25.0(0.14%),B:4.7(0.01%),P:1e-17\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t11 of 25 Initial Sequence: TAACCGTTTT... (-27.407)\n",
            "\t\tRound 1: -31.16 TAACCGTTTT T:1624.0(8.70%),B:2407.0(7.20%),P:1e-13\n",
            "\t\tRound 2: -31.16 TAACCGTTTT T:1624.0(8.70%),B:2407.0(7.20%),P:1e-13\n",
            "\t\t=Final=: -35.00 TAACCGTTTT T:1537.0(8.61%),B:2266.0(7.03%),P:1e-15\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t12 of 25 Initial Sequence: ACGTCCCCAG... (-24.794)\n",
            "\t\tRound 1: -33.24 ACGTCCCCAG T:27.0(0.15%),B:7.4(0.02%),P:1e-14\n",
            "\t\tRound 2: -33.24 ACGTCCCCAG T:27.0(0.15%),B:7.4(0.02%),P:1e-14\n",
            "\t\t=Final=: -31.71 ACGTCCCCAG T:27.0(0.15%),B:7.4(0.02%),P:1e-13\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t13 of 25 Initial Sequence: CAGCCACCGC... (-24.511)\n",
            "\t\tRound 1: -24.79 CAGCCACCGC T:15.0(0.08%),B:3.8(0.01%),P:1e-10\n",
            "\t\tRound 2: -25.98 MATCCACCGC T:25.0(0.14%),B:8.8(0.02%),P:1e-11\n",
            "\t\tRound 3: -25.98 MATCCACCGC T:25.0(0.14%),B:8.8(0.02%),P:1e-11\n",
            "\t\t=Final=: -25.04 MATCCACCGC T:25.0(0.14%),B:8.8(0.03%),P:1e-10\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t14 of 25 Initial Sequence: CTATCTACAC... (-24.511)\n",
            "\t\tRound 1: -24.51 CTATCTACAC T:12.0(0.07%),B:2.9(0.01%),P:1e-10\n",
            "\t\tRound 2: -24.51 CTATCTACAC T:12.0(0.07%),B:2.9(0.01%),P:1e-10\n",
            "\t\t=Final=: -19.78 CTATCTACAC T:12.0(0.07%),B:2.9(0.01%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t15 of 25 Initial Sequence: GAATGGTAGC... (-24.511)\n",
            "\t\tRound 1: -27.41 GAATGGTAGC T:16.0(0.09%),B:3.8(0.01%),P:1e-11\n",
            "\t\tRound 2: -30.08 GAWYGGTAGC T:17.0(0.10%),B:3.8(0.01%),P:1e-13\n",
            "\t\tRound 3: -30.08 GAWYGGTAGC T:17.0(0.10%),B:3.8(0.01%),P:1e-13\n",
            "\t\t=Final=: -26.44 GAWYGGTAGC T:17.0(0.10%),B:3.8(0.01%),P:1e-11\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t16 of 25 Initial Sequence: CATGTGATAC... (-24.511)\n",
            "\t\tRound 1: -24.51 CATGTGATAC T:12.0(0.07%),B:2.8(0.01%),P:1e-10\n",
            "\t\tRound 2: -24.51 CATGTGATAC T:12.0(0.07%),B:2.8(0.01%),P:1e-10\n",
            "\t\t=Final=: -19.78 CATGTGATAC T:12.0(0.07%),B:2.8(0.01%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t17 of 25 Initial Sequence: AGACACTAAC... (-24.511)\n",
            "\t\tRound 1: -24.51 AGACACTAAC T:12.0(0.07%),B:2.8(0.01%),P:1e-10\n",
            "\t\tRound 2: -24.51 AGACACTAAC T:12.0(0.07%),B:2.8(0.01%),P:1e-10\n",
            "\t\t=Final=: -19.78 AGACACTAAC T:12.0(0.07%),B:2.8(0.01%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t18 of 25 Initial Sequence: GAAGGTACAC... (-24.511)\n",
            "\t\tRound 1: -24.51 GAAGGTACAC T:12.0(0.07%),B:2.2(0.01%),P:1e-10\n",
            "\t\tRound 2: -24.51 GAAGGTACAC T:12.0(0.07%),B:2.2(0.01%),P:1e-10\n",
            "\t\t=Final=: -6.91 GAAGGTACAC T:6.0(0.03%),B:2.2(0.01%),P:1e-2\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t19 of 25 Initial Sequence: CCGCGTTGCA... (-23.048)\n",
            "\t\tRound 1: -29.27 CCGCGTTGCA T:1835.0(9.77%),B:2768.2(8.23%),P:1e-12\n",
            "\t\tRound 2: -29.27 CCGCGTTGCA T:1835.0(9.77%),B:2768.2(8.23%),P:1e-12\n",
            "\t\t=Final=: -28.77 CCGCGTTGCA T:1717.0(9.62%),B:2614.0(8.11%),P:1e-12\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t20 of 25 Initial Sequence: GGTTAAACCA... (-23.137)\n",
            "\t\tRound 1: -36.66 GGTTAAACCA T:1214.0(6.58%),B:1707.8(5.16%),P:1e-15\n",
            "\t\tRound 2: -36.66 GGTTAAACCA T:1214.0(6.58%),B:1707.8(5.16%),P:1e-15\n",
            "\t\t=Final=: -29.20 GGTTAAACCA T:1004.0(5.63%),B:1437.6(4.46%),P:1e-12\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t21 of 25 Initial Sequence: CAATGGTCCT... (-22.251)\n",
            "\t\tRound 1: -24.11 CAATGGTCCT T:17.0(0.10%),B:4.6(0.01%),P:1e-10\n",
            "\t\tRound 2: -24.18 SAATSGTCCT T:19.0(0.11%),B:5.6(0.02%),P:1e-10\n",
            "\t\tRound 3: -24.18 SAATSGTCCT T:19.0(0.11%),B:5.6(0.02%),P:1e-10\n",
            "\t\t=Final=: -22.61 SAATSGTCCT T:19.0(0.11%),B:5.6(0.02%),P:1e-9\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t22 of 25 Initial Sequence: CCTCCCTAAG... (-22.221)\n",
            "\t\tRound 1: -43.09 CCTCCCTAAG T:2450.0(12.83%),B:3646.1(10.70%),P:1e-18\n",
            "\t\tRound 2: -45.60 CCTCCCTAAG T:2215.0(11.67%),B:3244.9(9.58%),P:1e-19\n",
            "\t\tRound 3: -45.60 CCTCCCTAAG T:2215.0(11.67%),B:3244.9(9.58%),P:1e-19\n",
            "\t\t=Final=: -44.77 CCTCCCTAAG T:2069.0(11.59%),B:3069.2(9.53%),P:1e-19\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t23 of 25 Initial Sequence: GGAAACCACT... (-23.147)\n",
            "\t\tRound 1: -46.74 GGAAACCACT T:314.0(1.74%),B:318.0(0.98%),P:1e-20\n",
            "\t\tRound 2: -46.74 GGAAACCACT T:314.0(1.74%),B:318.0(0.98%),P:1e-20\n",
            "\t\t=Final=: -46.74 GGAAACCACT T:311.0(1.74%),B:315.8(0.98%),P:1e-20\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t24 of 25 Initial Sequence: TTAGCTTGTT... (-22.073)\n",
            "\t\tRound 1: -25.52 TTAGCTTGTT T:38.0(0.21%),B:18.6(0.06%),P:1e-11\n",
            "\t\tRound 2: -25.52 TTAGCTTGTT T:38.0(0.21%),B:18.6(0.06%),P:1e-11\n",
            "\t\t=Final=: -25.28 TTAGCTTGTT T:38.0(0.21%),B:18.6(0.06%),P:1e-10\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t25 of 25 Initial Sequence: GTGCCCTTTC... (-21.839)\n",
            "\t\tRound 1: -28.83 GTGCCCTTTC T:19.0(0.11%),B:4.5(0.01%),P:1e-12\n",
            "\t\tRound 2: -33.95 GCGCCCTTYC T:34.0(0.19%),B:11.0(0.03%),P:1e-14\n",
            "\t\tRound 3: -33.95 GCGCCCTTYC T:34.0(0.19%),B:11.0(0.03%),P:1e-14\n",
            "\t\t=Final=: -33.06 GCGCCCTTYC T:34.0(0.19%),B:11.0(0.03%),P:1e-14\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\n",
            "\tFinalizing Enrichment Statistics (new in v3.4)\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\tChecking enrichment of 25 motif(s)\n",
            "\t|0%                                    50%                                  100%|\n",
            "\t=================================================================================\n",
            "\tOutput in file: EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF/homerMotifs.motifs10\n",
            "\n\n",
            "\t-blen automatically set to 2\n",
            "\tScanning input files...\n",
            "\tParsing sequences...\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t===============================================================================\n",
            "\tTotal number of Oligos: 3802138\n",
            "\tAutoadjustment for sequence coverage in background: 1.00x\n",
            "\n",
            "\tOligos: 3802138 of 8830118 max\n",
            "\tTree  : 8640492 of 44150590 max\n",
            "\tOptimizing memory usage...\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\n",
            "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
            "\n",
            "\tScreening oligos 3802138 (allowing 0 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t42.68% skipped, 57.32% checked (2179242 of 3802138), of those checked:\n",
            "\t\t42.68% not in target, 0.00% increased p-value, 0.00% high p-value\n",
            "\n",
            "\tScreening oligos 3802138 (allowing 1 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t42.68% skipped, 57.32% checked (2179242 of 3802138), of those checked:\n",
            "\t\t0.00% not in target, 29.17% increased p-value, 35.04% high p-value\n",
            "\n",
            "\tScreening oligos 3802138 (allowing 2 mismatches):\n",
            "\t|0%                                   50%                                  100%|\n",
            "\t================================================================================\n",
            "\t\t80.99% skipped, 19.01% checked (722638 of 3802138), of those checked:\n",
            "\t\t0.00% not in target, 13.83% increased p-value, 0.01% high p-value\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\n",
            "\tLocal Optimization Phase:\n",
            "\t1 of 25 Initial Sequence: ACCCTAACCCTA... (-174.122)\n",
            "\t\tRound 1: -220.03 ACCCTAACCCTA T:60.0(0.34%),B:2.8(0.01%),P:1e-95\n",
            "\t\tRound 2: -220.03 ACCCTAACCCTA T:60.0(0.34%),B:2.8(0.01%),P:1e-95\n",
            "\t\t=Final=: -17.39 ACCCTAACCCTA T:11.0(0.06%),B:2.8(0.01%),P:1e-7\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t2 of 25 Initial Sequence: ACTACACTATGG... (-47.562)\n",
            "\t\tRound 1: -58.33 ACTACACTATGG T:22.0(0.12%),B:2.2(0.01%),P:1e-25\n",
            "\t\tRound 2: -65.74 ACTACACTATGG T:24.0(0.13%),B:2.2(0.01%),P:1e-28\n",
            "\t\tRound 3: -65.74 ACTACACTATGG T:24.0(0.13%),B:2.2(0.01%),P:1e-28\n",
            "\t\t=Final=: -47.29 ACTACACTATGG T:22.0(0.12%),B:2.2(0.01%),P:1e-20\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t3 of 25 Initial Sequence: TCCTCCGAATGC... (-44.078)\n",
            "\t\tRound 1: -62.01 TCCTCCGAATGC T:23.0(0.13%),B:1.7(0.00%),P:1e-26\n",
            "\t\tRound 2: -62.01 TCCTCCGAATGC T:23.0(0.13%),B:1.7(0.00%),P:1e-26\n",
            "\t\t=Final=: -65.74 TCCTCCGAATGC T:23.0(0.13%),B:1.7(0.01%),P:1e-28\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t4 of 25 Initial Sequence: CAAACCAAACCA... (-41.924)\n",
            "\t\tRound 1: -60.52 CAAACCAAACCA T:174.0(0.97%),B:123.4(0.38%),P:1e-26\n",
            "\t\tRound 2: -63.16 CAAACCAAACCA T:185.0(1.03%),B:132.7(0.41%),P:1e-27\n",
            "\t\tRound 3: -63.48 CAAACCAAACCA T:178.0(0.99%),B:124.7(0.38%),P:1e-27\n",
            "\t\tRound 4: -63.55 CAAACCAAACCA T:179.0(1.00%),B:125.5(0.39%),P:1e-27\n",
            "\t\tRound 5: -63.55 CAAACCAAACCA T:179.0(1.00%),B:125.5(0.39%),P:1e-27\n",
            "\t\t=Final=: -20.08 CAAACCAAACCA T:90.0(0.50%),B:81.4(0.25%),P:1e-8\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t5 of 25 Initial Sequence: GTGTGAACCGGC... (-40.652)\n",
            "\t\tRound 1: -47.56 GTGTGAACCGGC T:19.0(0.11%),B:2.0(0.01%),P:1e-20\n",
            "\t\tRound 2: -47.56 GTGTGAACCGGC T:19.0(0.11%),B:2.0(0.01%),P:1e-20\n",
            "\t\t=Final=: -31.27 GTGTGAACCGGC T:19.0(0.11%),B:3.2(0.01%),P:1e-13\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t6 of 25 Initial Sequence: GTGCTGTGCTGT... (-35.208)\n",
            "\t\tRound 1: -45.52 GTGCTGTGCTGT T:45.0(0.25%),B:14.5(0.04%),P:1e-19\n",
            "\t\tRound 2: -49.22 GTGCTGTGCTGT T:32.0(0.18%),B:7.0(0.02%),P:1e-21\n",
            "\t\tRound 3: -49.81 GTGCTGTGCTGT T:44.0(0.25%),B:12.7(0.04%),P:1e-21\n",
            "\t\tRound 4: -49.81 GTGCTGTGCTGT T:44.0(0.25%),B:12.7(0.04%),P:1e-21\n",
            "\t\t=Final=: -15.68 GTGCTGTGCTGT T:24.0(0.13%),B:12.7(0.04%),P:1e-6\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t7 of 25 Initial Sequence: GTCTACATTCTC... (-34.974)\n",
            "\t\tRound 1: -79.38 GTCTACATTCTC T:33.0(0.18%),B:3.5(0.01%),P:1e-34\n",
            "\t\tRound 2: -82.78 GTCTACATTCTC T:34.0(0.19%),B:3.5(0.01%),P:1e-35\n",
            "\t\tRound 3: -82.78 GTCTACATTCTC T:34.0(0.19%),B:3.5(0.01%),P:1e-35\n",
            "\t\t=Final=: -72.96 GTCTACATTCTC T:34.0(0.19%),B:3.5(0.01%),P:1e-31\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t8 of 25 Initial Sequence: GTTAATATTGCA... (-33.983)\n",
            "\t\tRound 1: -47.56 GTTAATATTGCA T:19.0(0.11%),B:2.9(0.01%),P:1e-20\n",
            "\t\tRound 2: -47.56 GTTAATATTGCA T:19.0(0.11%),B:2.9(0.01%),P:1e-20\n",
            "\t\t=Final=: -42.56 GTTAATATTGCA T:77.0(0.43%),B:42.0(0.13%),P:1e-18\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t9 of 25 Initial Sequence: GGATACACGCCC... (-33.983)\n",
            "\t\tRound 1: -51.10 GGATACACGCCC T:20.0(0.11%),B:2.2(0.01%),P:1e-22\n",
            "\t\tRound 2: -54.69 GGATACACGCCC T:21.0(0.12%),B:2.2(0.01%),P:1e-23\n",
            "\t\tRound 3: -54.69 GGATACACGCCC T:21.0(0.12%),B:2.2(0.01%),P:1e-23\n",
            "\t\t=Final=: -44.30 GGATACACGCCC T:21.0(0.12%),B:2.2(0.01%),P:1e-19\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t10 of 25 Initial Sequence: TCCAGTGAACAA... (-32.162)\n",
            "\t\tRound 1: -47.56 TCCAGTGAACAA T:19.0(0.11%),B:0.5(0.00%),P:1e-20\n",
            "\t\tRound 2: -66.11 TCCAGTGAACAA T:29.0(0.16%),B:4.0(0.01%),P:1e-28\n",
            "\t\tRound 3: -84.99 TCCAGTGAACAA T:29.0(0.16%),B:2.1(0.01%),P:1e-36\n",
            "\t\tRound 4: -86.21 TCCAGTGAACAA T:35.0(0.20%),B:3.2(0.01%),P:1e-37\n",
            "\t\tRound 5: -91.67 YMCAGTGRACAA T:41.0(0.23%),B:4.8(0.01%),P:1e-39\n",
            "\t\tRound 6: -91.67 YMCAGTGRACAA T:41.0(0.23%),B:4.8(0.01%),P:1e-39\n",
            "\t\t=Final=: -83.62 YMCAGTGRACAA T:41.0(0.23%),B:4.8(0.01%),P:1e-36\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t11 of 25 Initial Sequence: CATGTGTGTGTA... (-32.156)\n",
            "\t\tRound 1: -41.13 CATGTGTGTGTA T:51.0(0.29%),B:20.5(0.06%),P:1e-17\n",
            "\t\tRound 2: -41.13 CATGTGTGTGTA T:51.0(0.29%),B:20.5(0.06%),P:1e-17\n",
            "\t\t=Final=: -21.10 CATGTGTGTGTA T:37.0(0.21%),B:20.5(0.06%),P:1e-9\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t12 of 25 Initial Sequence: TTGATAGTAATG... (-31.275)\n",
            "\t\tRound 1: -44.12 TTGATAGTAATG T:34.0(0.19%),B:8.8(0.02%),P:1e-19\n",
            "\t\tRound 2: -44.12 TTGATAGTAATG T:34.0(0.19%),B:8.8(0.02%),P:1e-19\n",
            "\t\t=Final=: -42.29 TTGATAGTAATG T:34.0(0.19%),B:8.8(0.03%),P:1e-18\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t13 of 25 Initial Sequence: CTACCATTTAGT... (-30.794)\n",
            "\t\tRound 1: -57.36 CTACCATTTAGT T:44.0(0.25%),B:10.1(0.03%),P:1e-24\n",
            "\t\tRound 2: -57.36 CTACCATTTAGT T:44.0(0.25%),B:10.1(0.03%),P:1e-24\n",
            "\t\t=Final=: -55.45 CTACCATTTAGT T:44.0(0.25%),B:10.1(0.03%),P:1e-24\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t14 of 25 Initial Sequence: ACTTTGACGTTA... (-30.750)\n",
            "\t\tRound 1: -33.98 ACTTTGACGTTA T:15.0(0.08%),B:2.8(0.01%),P:1e-14\n",
            "\t\tRound 2: -44.08 ACTTTGACGTTA T:18.0(0.10%),B:2.9(0.01%),P:1e-19\n",
            "\t\tRound 3: -44.08 ACTTTGACGTTA T:18.0(0.10%),B:2.9(0.01%),P:1e-19\n",
            "\t\t=Final=: -40.65 ACTTTGACGTTA T:16.0(0.09%),B:1.7(0.01%),P:1e-17\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t15 of 25 Initial Sequence: ACGTGTAATCAA... (-30.750)\n",
            "\t\tRound 1: -33.98 ACGTGTAATCAA T:15.0(0.08%),B:2.1(0.01%),P:1e-14\n",
            "\t\tRound 2: -33.98 ACGTGTAATCAA T:15.0(0.08%),B:2.1(0.01%),P:1e-14\n",
            "\t\t=Final=: -33.98 ACGTGTAATCAA T:14.0(0.08%),B:0.7(0.00%),P:1e-14\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t16 of 25 Initial Sequence: ATCCCGAACAGT... (-30.750)\n",
            "\t\tRound 1: -33.77 ATCCCGAACAGT T:21.0(0.12%),B:4.9(0.01%),P:1e-14\n",
            "\t\tRound 2: -58.33 ATCCWGAACAGT T:22.0(0.12%),B:2.7(0.01%),P:1e-25\n",
            "\t\tRound 3: -58.33 ATCCWGAACAGT T:22.0(0.12%),B:2.7(0.01%),P:1e-25\n",
            "\t\t=Final=: -54.69 ATCCWGAACAGT T:20.0(0.11%),B:1.1(0.00%),P:1e-23\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t17 of 25 Initial Sequence: GTCCACGAGTGT... (-30.750)\n",
            "\t\tRound 1: -30.75 GTCCACGAGTGT T:14.0(0.08%),B:2.7(0.01%),P:1e-13\n",
            "\t\tRound 2: -30.75 GTCCACGAGTGT T:14.0(0.08%),B:2.7(0.01%),P:1e-13\n",
            "\t\t=Final=: -24.79 GTCCACGAGTGT T:14.0(0.08%),B:2.7(0.01%),P:1e-10\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t18 of 25 Initial Sequence: GACAAGTGAAAG... (-30.327)\n",
            "\t\tRound 1: -56.52 GACAAGTGAAAG T:26.0(0.15%),B:3.9(0.01%),P:1e-24\n",
            "\t\tRound 2: -69.51 GACAAGTGAAAG T:25.0(0.14%),B:1.8(0.00%),P:1e-30\n",
            "\t\tRound 3: -69.51 GACAAGTGAAAG T:25.0(0.14%),B:1.8(0.00%),P:1e-30\n",
            "\t\t=Final=: -73.32 GACAAGTGAAAG T:25.0(0.14%),B:1.8(0.01%),P:1e-31\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t19 of 25 Initial Sequence: ACACACTGCTTC... (-30.288)\n",
            "\t\tRound 1: -45.85 ACACACTGCTTC T:370.0(2.05%),B:395.2(1.22%),P:1e-19\n",
            "\t\tRound 2: -45.85 ACACACTGCTTC T:370.0(2.05%),B:395.2(1.22%),P:1e-19\n",
            "\t\t=Final=: -40.66 ACACACTGCTTC T:530.0(2.97%),B:643.5(2.00%),P:1e-17\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t20 of 25 Initial Sequence: GGAAATAGCAGT... (-31.658)\n",
            "\t\tRound 1: -52.99 GGAAATAGCAGT T:38.0(0.21%),B:8.3(0.02%),P:1e-23\n",
            "\t\tRound 2: -53.36 GGAAATAGCAGT T:36.0(0.20%),B:7.5(0.02%),P:1e-23\n",
            "\t\tRound 3: -62.88 GGAAATAGCAGT T:28.0(0.16%),B:3.3(0.01%),P:1e-27\n",
            "\t\tRound 4: -69.51 GGAAATAGCAGT T:25.0(0.14%),B:2.2(0.01%),P:1e-30\n",
            "\t\tRound 5: -69.51 GGAAATAGCAGT T:25.0(0.14%),B:2.2(0.01%),P:1e-30\n",
            "\t\t=Final=: -56.52 GGAAATAGCAGT T:25.0(0.14%),B:2.2(0.01%),P:1e-24\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t21 of 25 Initial Sequence: AGTGGTGCCGAG... (-30.081)\n",
            "\t\tRound 1: -38.39 AGTGGTGCCGAG T:41.0(0.23%),B:14.7(0.04%),P:1e-16\n",
            "\t\tRound 2: -38.39 AGTGGTGCCGAG T:41.0(0.23%),B:14.7(0.04%),P:1e-16\n",
            "\t\t=Final=: -35.96 AGTGGTGCCGAG T:40.0(0.22%),B:14.7(0.05%),P:1e-15\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t22 of 25 Initial Sequence: GCATTTGACGCT... (-28.832)\n",
            "\t\tRound 1: -51.10 GCATTTGASGCT T:20.0(0.11%),B:2.1(0.01%),P:1e-22\n",
            "\t\tRound 2: -51.10 GCATTTGASGCT T:20.0(0.11%),B:2.1(0.01%),P:1e-22\n",
            "\t\t=Final=: -41.35 GCATTTGASGCT T:20.0(0.11%),B:2.1(0.01%),P:1e-17\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t23 of 25 Initial Sequence: AAACCACGTCAC... (-28.832)\n",
            "\t\tRound 1: -47.56 AAACCACGTCAC T:19.0(0.11%),B:2.3(0.01%),P:1e-20\n",
            "\t\tRound 2: -47.56 AAACCACGTCAC T:19.0(0.11%),B:2.3(0.01%),P:1e-20\n",
            "\t\t=Final=: -30.08 AAACCACGTCAC T:16.0(0.09%),B:2.3(0.01%),P:1e-13\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t24 of 25 Initial Sequence: ACGCTCAGTCAA... (-27.591)\n",
            "\t\tRound 1: -33.98 ACGCTCAGTCAA T:15.0(0.08%),B:2.7(0.01%),P:1e-14\n",
            "\t\tRound 2: -33.98 ACGCTCAGTCAA T:15.0(0.08%),B:2.7(0.01%),P:1e-14\n",
            "\t\t=Final=: -27.41 ACGCTCAGTCAA T:15.0(0.08%),B:2.7(0.01%),P:1e-11\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\t25 of 25 Initial Sequence: TGTAATCGAGCA... (-27.591)\n",
            "\t\tRound 1: -28.64 TGTAATCGAGCA T:28.0(0.16%),B:9.9(0.03%),P:1e-12\n",
            "\t\tRound 2: -30.75 TGTAATCGAGCA T:14.0(0.08%),B:3.0(0.01%),P:1e-13\n",
            "\t\tRound 3: -30.75 TGTAATCGAGCA T:14.0(0.08%),B:3.0(0.01%),P:1e-13\n",
            "\t\t=Final=: -27.59 TGTAATCGAGCA T:12.0(0.07%),B:0.8(0.00%),P:1e-11\n",
            "\t\tPerforming exhaustive masking of motif...\n",
            "\t\tReprioritizing potential motifs...\n",
            "\n",
            "\tFinalizing Enrichment Statistics (new in v3.4)\n",
            "\tReading input files...\n",
            "\t50000 total sequences read\n",
            "\tCache length = 11180\n",
            "\tUsing binomial scoring\n",
            "\tChecking enrichment of 25 motif(s)\n",
            "\t|0%                                    50%                                  100%|\n",
            "\t=================================================================================\n",
            "\tOutput in file: EFF_H3K4Me1_1_srt_dupr_chr11_HomerMF/homerMotifs.motifs12\n",
            "\n",
            "\t(Motifs in homer2 format)\n",
            "\tDetermining similar motifs... 75 reduced to 50 motifs\n",
            "\tOutputing HTML and sequence logos for motif comparison...\n",
            "\tChecking de novo motifs against known motifs...\n",
            "\tFormatting HTML page...\n",
            "\t\t1 of 50 (1e-36) similar to eyg/dmmpmm(Bergman)/fly(0.656)\n",
            "\t\t2 of 50 (1e-31) similar to FMR1(KH)/Homo_sapiens-RNCMPT00016-PBM/HughesRNA(0.712)\n",
            "\t\t3 of 50 (1e-31) similar to TEC1(MacIsaac)/Yeast(0.707)\n",
            "\t\t4 of 50 (1e-28) similar to LIN28A(CSD)/Homo_sapiens-RNCMPT00162-PBM/HughesRNA(0.714)\n",
            "\t\t5 of 50 (1e-25) similar to FZF1/MA0298.1/Jaspar(0.667)\n",
            "\t\t6 of 50 (1e-24) similar to YAP3/MA0416.1/Jaspar(0.804)\n",
            "\t\t7 of 50 (1e-24) similar to BARHL2/MA0635.1/Jaspar(0.702)\n",
            "\t\t8 of 50 (1e-23) similar to Ot_0263(RRM)/Ostreococcus_tauri-RNCMPT00263-PBM/HughesRNA(0.608)\n",
            "\t\t9 of 50 (1e-21) similar to HNRNPA2B1(RRM)/Homo_sapiens-RNCMPT00024-PBM/HughesRNA(0.790)\n",
            "\t\t10 of 50 (1e-20) similar to ASCL1/MA1100.1/Jaspar(0.710)\n",
            "\t\t11 of 50 (1e-20) similar to Tb_0216(RRM)/Trypanosoma_brucei-RNCMPT00216-PBM/HughesRNA(0.738)\n",
            "\t\t12 of 50 (1e-20) similar to prd/dmmpmm(Down)/fly(0.825)\n",
            "\t\t13 of 50 (1e-20) similar to HAP1/HAP1_YPD/[](Harbison)/Yeast(0.690)\n",
            "\t\t14 of 50 (1e-19) similar to vnd/dmmpmm(Bergman)/fly(0.740)\n",
            "\t\t15 of 50 (1e-19) similar to KLF10(Zf)/HEK293-KLF10.GFP-ChIP-Seq(GSE58341)/Homer(0.692)\n",
            "\t\t16 of 50 (1e-18) similar to Dof2/MA0020.1/Jaspar(0.806)\n",
            "\t\t17 of 50 (1e-18) similar to Arid5a/MA0602.1/Jaspar(0.741)\n",
            "\t\t18 of 50 (1e-18) similar to TBP3(MYBrelated)/col-TBP3-DAP-Seq(GSE60143)/Homer(0.739)\n",
            "\t\t19 of 50 (1e-17) similar to CST6(MacIsaac)/Yeast(0.678)\n",
            "\t\t20 of 50 (1e-17) similar to AZF1(C2H2)/colamp-AZF1-DAP-Seq(GSE60143)/Homer(0.733)\n",
            "\t\t21 of 50 (1e-17) similar to che-1/MA0260.1/Jaspar(0.743)\n",
            "\t\t22 of 50 (1e-17) similar to Tcf7(HMG)/GM12878-TCF7-ChIP-Seq(Encode)/Homer(0.755)\n",
            "\t\t23 of 50 (1e-17) similar to GATA20/MA1324.1/Jaspar(0.780)\n",
            "\t\t24 of 50 (1e-15) similar to Pr_0249(RRM)/Phytophthora_ramorum-RNCMPT00249-PBM/HughesRNA(0.692)\n",
            "\t\t25 of 50 (1e-15) similar to ARALYDRAFT_897773/MA1054.1/Jaspar(0.699)\n",
            "\t\t26 of 50 (1e-15) similar to MYB.PH3(1)(MYB)/Petunia hybrida/AthaMap(0.913)\n",
            "\t\t27 of 50 (1e-14) similar to MATALPHA2/MA0328.2/Jaspar(0.750)\n",
            "\t\t28 of 50 (1e-13) similar to Su(H)/dmmpmm(Papatsenko)/fly(0.705)\n",
            "\t\t29 of 50 (1e-12) similar to CMTA3/MA0970.1/Jaspar(0.739)\n",
            "\t\t30 of 50 (1e-12) similar to MYB119/MA1176.1/Jaspar(0.822)\n",
            "\t\t31 of 50 (1e-12) similar to MZF1/MA0056.1/Jaspar(0.703)\n",
            "\t\t32 of 50 (1e-11) similar to ZAP1/MA0589.1/Jaspar(0.723)\n",
            "\t\t33 of 50 (1e-10) similar to CHA4/MA0283.1/Jaspar(0.832)\n",
            "\t\t34 of 50 (1e-10) similar to XBP1(MacIsaac)/Yeast(0.638)\n",
            "\t\t35 of 50 (1e-10) similar to STP3/MA0396.1/Jaspar(0.782)\n",
            "\t\t36 of 50 (1e-9) similar to DAG2/MA1271.1/Jaspar(0.713)\n",
            "\t\t37 of 50 (1e-9) similar to SRSF1(RRM)/Homo_sapiens-RNCMPT00110-PBM/HughesRNA(0.699)\n",
            "\t\t38 of 50 (1e-9) similar to SeqBias: CA-repeat(0.776)\n",
            "\t\t39 of 50 (1e-8) similar to AT3G57600(AP2EREBP)/col-AT3G57600-DAP-Seq(GSE60143)/Homer(0.870)\n",
            "\t\t40 of 50 (1e-8) similar to HuR(RRM)/Homo_sapiens-RNCMPT00136-PBM/HughesRNA(0.672)\n",
            "\t\t41 of 50 (1e-8) similar to HNRNPAB(RRM)/Tetraodon_nigroviridis-RNCMPT00245-PBM/HughesRNA(0.803)\n",
            "\t\t42 of 50 (1e-8) similar to TCX2(CPP)/colamp-TCX2-DAP-Seq(GSE60143)/Homer(0.794)\n",
            "\t\t43 of 50 (1e-7) similar to ZNF711(Zf)/SHSY5Y-ZNF711-ChIP-Seq(GSE20673)/Homer(0.851)\n",
            "\t\t44 of 50 (1e-7) similar to MUB(KH)/Drosophila_melanogaster-RNCMPT00137-PBM/HughesRNA(0.728)\n",
            "\t\t45 of 50 (1e-7) similar to PUCHI(AP2EREBP)/colamp-PUCHI-DAP-Seq(GSE60143)/Homer(0.819)\n",
            "\t\t46 of 50 (1e-6) similar to PB0091.1_Zbtb3_1/Jaspar(0.656)\n",
            "\t\t47 of 50 (1e-6) similar to Deaf1/dmmpmm(Bigfoot)/fly(0.846)\n",
            "\t\t48 of 50 (1e-5) similar to PB0009.1_E2F3_1/Jaspar(0.680)\n",
            "\t\t49 of 50 (1e-5) similar to byn/dmmpmm(Pollard)/fly(0.749)\n",
            "\t\t50 of 50 (1e-4) similar to SF1(KH)/Drosophila_melanogaster-RNCMPT00065-PBM/HughesRNA(0.704)\n",
            "\tJob finished - if results look good, please send beer to ..\n",
            "\n",
            "\tCleaning up tmp files...\n",
            "\n"
          ]
        }
      ],
      "execution_count": 8,
      "metadata": {}
    },
    {
      "cell_type": "code",
      "source": [],
      "outputs": [],
      "execution_count": null,
      "metadata": {
        "collapsed": false,
        "outputHidden": false,
        "inputHidden": false
      }
    }
  ],
  "metadata": {
    "kernel_info": {
      "name": "python3"
    },
    "language_info": {
      "name": "python",
      "version": "3.7.1",
      "mimetype": "text/x-python",
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "pygments_lexer": "ipython3",
      "nbconvert_exporter": "python",
      "file_extension": ".py"
    },
    "kernelspec": {
      "name": "python3",
      "language": "python",
      "display_name": "Python 3"
    },
    "nteract": {
      "version": "0.12.3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 4
}